{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MASH Analysis for Finished and NCBI Cyanobacterial Genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Outline\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Importing General Dependencies**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import glob\n",
    "import subprocess\n",
    "import os\n",
    "import time\n",
    "import re\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "sns.set()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Installing MASH**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MASH already downloaded\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"./mash-OSX64-v2.2/mash\"):\n",
    "    print(\"MASH already downloaded\")\n",
    "else:\n",
    "    !wget https://github.com/marbl/Mash/releases/download/v2.2/mash-OSX64-v2.2.tar\n",
    "    !tar xopf mash-OSX64-v2.2.tar; rm mash-OSX64-v2.2.tar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Sketching Finished Cyanobiome and NCBI Datasets**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This step is going to create the minhashes from the references to be probed in the soon to be queried genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sketch already exists\n",
      "\n",
      "--- 0.00020003318786621094 seconds ---\n"
     ]
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "\n",
    "if os.path.exists(\"./mash-OSX64-v2.2/ncbi_genomes.msh\"):\n",
    "    print(\"Sketch already exists\")\n",
    "else:\n",
    "    !./mash-OSX64-v2.2/mash sketch -o ./mash-OSX64-v2.2/ncbi_genomes ./ncbi_genomes/*.fasta\n",
    "    \n",
    "print('\\n' + \"--- %s seconds ---\" %(time.time()-start_time))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: ./query_genomes/: File exists\r\n"
     ]
    }
   ],
   "source": [
    "query_genomes = ['JAAHII01','JAAHIH01','JAAHGF01','JAAHHC01','JAAHHD01','JAAHFN01','JAAHGM01','JAAHGB01','JAAHFW01','JAAHFZ01','JAAHFP01','JAAHGH01','JAAHGC01','JAAHGA01','JAAHHJ01','JAAHGO01','JAAHGU01','JAAHGW01','JAAHGS01','JAAHHM01','JAAHHP01','JAAHHN01','JAAHGL01','JAAHFU01']\n",
    "\n",
    "!mkdir ./query_genomes/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "glob_list = glob.glob('./ncbi_genomes/*fasta')\n",
    "\n",
    "for item in glob_list:\n",
    "    name = os.path.basename(item).split('.')[0]\n",
    "    if name in query_genomes:\n",
    "        cmd = 'cp %s ./query_genomes/'%item\n",
    "        subprocess.call(cmd,shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./query_genomes/JAAHHP01.fasta ./ncbi_genomes/JAAHHP01.fasta\n",
      "./query_genomes/JAAHGA01.fasta ./ncbi_genomes/JAAHGA01.fasta\n",
      "./query_genomes/JAAHGH01.fasta ./ncbi_genomes/JAAHGH01.fasta\n",
      "./query_genomes/JAAHHM01.fasta ./ncbi_genomes/JAAHHM01.fasta\n",
      "./query_genomes/JAAHHD01.fasta ./ncbi_genomes/JAAHHD01.fasta\n",
      "./query_genomes/JAAHGU01.fasta ./ncbi_genomes/JAAHGU01.fasta\n",
      "./query_genomes/JAAHFU01.fasta ./ncbi_genomes/JAAHFU01.fasta\n",
      "./query_genomes/JAAHGB01.fasta ./ncbi_genomes/JAAHGB01.fasta\n",
      "./query_genomes/JAAHFW01.fasta ./ncbi_genomes/JAAHFW01.fasta\n",
      "./query_genomes/JAAHGW01.fasta ./ncbi_genomes/JAAHGW01.fasta\n",
      "./query_genomes/JAAHGC01.fasta ./ncbi_genomes/JAAHGC01.fasta\n",
      "./query_genomes/JAAHHN01.fasta ./ncbi_genomes/JAAHHN01.fasta\n",
      "./query_genomes/JAAHFZ01.fasta ./ncbi_genomes/JAAHFZ01.fasta\n",
      "./query_genomes/JAAHGO01.fasta ./ncbi_genomes/JAAHGO01.fasta\n",
      "./query_genomes/JAAHGF01.fasta ./ncbi_genomes/JAAHGF01.fasta\n",
      "./query_genomes/JAAHGS01.fasta ./ncbi_genomes/JAAHGS01.fasta\n",
      "./query_genomes/JAAHHC01.fasta ./ncbi_genomes/JAAHHC01.fasta\n",
      "./query_genomes/JAAHFN01.fasta ./ncbi_genomes/JAAHFN01.fasta\n",
      "./query_genomes/JAAHHJ01.fasta ./ncbi_genomes/JAAHHJ01.fasta\n",
      "./query_genomes/JAAHFP01.fasta ./ncbi_genomes/JAAHFP01.fasta\n",
      "./query_genomes/JAAHGL01.fasta ./ncbi_genomes/JAAHGL01.fasta\n",
      "./query_genomes/JAAHIH01.fasta ./ncbi_genomes/JAAHIH01.fasta\n",
      "./query_genomes/JAAHGM01.fasta ./ncbi_genomes/JAAHGM01.fasta\n",
      "./query_genomes/JAAHII01.fasta ./ncbi_genomes/JAAHII01.fasta\n",
      "\n",
      "--- 11.281111001968384 seconds ---\n"
     ]
    }
   ],
   "source": [
    "def get_mash_df(inputs_folder,sketch):\n",
    "    start_time = time.time()\n",
    "    input_list = glob.glob(\"%s*.fasta\"%inputs_folder)\n",
    "    final_df = pd.DataFrame()\n",
    "    for index,item in enumerate(input_list):\n",
    "        scores = []\n",
    "        ncbi_item = './ncbi_genomes/%s'%(os.path.basename(item))\n",
    "        print(item,ncbi_item)\n",
    "        cmd = \"./mash-OSX64-v2.2/mash dist %s %s > ./mash-OSX64-v2.2/temp.txt\"%(sketch,item)\n",
    "        subprocess.call(cmd,shell=True)\n",
    "        df = pd.read_csv(\"./mash-OSX64-v2.2/temp.txt\",sep=\"\\t\",names=[\"reference\",\"query\",\"distance\",\"pvalue\",\"matching-hashes\"])\n",
    "        df = df[df.reference != ncbi_item]\n",
    "        row = df[df.distance == df.distance.min()]\n",
    "        final_df = final_df.append(row[:1], ignore_index = True)\n",
    "    print('\\n' + \"--- %s seconds ---\" %(time.time()-start_time))\n",
    "    return final_df\n",
    "        \n",
    "mash_df = get_mash_df(\"./query_genomes/\",\n",
    "                      \"./mash-OSX64-v2.2/ncbi_genomes.msh\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reference</th>\n",
       "      <th>query</th>\n",
       "      <th>distance</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>matching-hashes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./ncbi_genomes/JAAHHM01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHP01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>./ncbi_genomes/JAAHGB01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGA01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>./ncbi_genomes/JAAHGF01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGH01.fasta</td>\n",
       "      <td>0.041285</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>266/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>./ncbi_genomes/JAAHHP01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHM01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>./ncbi_genomes/AEPQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHD01.fasta</td>\n",
       "      <td>0.000781</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>968/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>./ncbi_genomes/JAAHGW01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGU01.fasta</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>992/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>./ncbi_genomes/JAAHFR01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFU01.fasta</td>\n",
       "      <td>0.219531</td>\n",
       "      <td>5.955950e-18</td>\n",
       "      <td>5/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>./ncbi_genomes/JAAHGA01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGB01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>./ncbi_genomes/RCBZ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFW01.fasta</td>\n",
       "      <td>0.004959</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>820/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>./ncbi_genomes/JAAHGU01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGW01.fasta</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>992/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>./ncbi_genomes/JAAHGB01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGC01.fasta</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>983/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>./ncbi_genomes/ALVQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHN01.fasta</td>\n",
       "      <td>0.230110</td>\n",
       "      <td>8.777560e-15</td>\n",
       "      <td>4/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>./ncbi_genomes/JAAHHC01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFZ01.fasta</td>\n",
       "      <td>0.016275</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>551/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>./ncbi_genomes/JAAHGP01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGO01.fasta</td>\n",
       "      <td>0.051530</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>204/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>./ncbi_genomes/JAAHHR01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGF01.fasta</td>\n",
       "      <td>0.013749</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>599/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>./ncbi_genomes/JAAHFQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGS01.fasta</td>\n",
       "      <td>0.098138</td>\n",
       "      <td>1.157750e-306</td>\n",
       "      <td>68/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>./ncbi_genomes/JAAHHD01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHC01.fasta</td>\n",
       "      <td>0.007596</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>743/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>./ncbi_genomes/RFFC01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFN01.fasta</td>\n",
       "      <td>0.230110</td>\n",
       "      <td>2.063740e-14</td>\n",
       "      <td>4/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>./ncbi_genomes/JAAHFQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHJ01.fasta</td>\n",
       "      <td>0.008967</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>707/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>./ncbi_genomes/JAAHFO01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFP01.fasta</td>\n",
       "      <td>0.020280</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>485/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>./ncbi_genomes/JAAHFM01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGL01.fasta</td>\n",
       "      <td>0.074432</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>117/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>./ncbi_genomes/JAAHHQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHIH01.fasta</td>\n",
       "      <td>0.024841</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>422/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>./ncbi_genomes/JAAHGB01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGM01.fasta</td>\n",
       "      <td>0.034699</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>318/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>./ncbi_genomes/QVFV01.fasta</td>\n",
       "      <td>./query_genomes/JAAHII01.fasta</td>\n",
       "      <td>0.243761</td>\n",
       "      <td>7.617380e-11</td>\n",
       "      <td>3/1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        reference                           query  distance  \\\n",
       "0   ./ncbi_genomes/JAAHHM01.fasta  ./query_genomes/JAAHHP01.fasta  0.000240   \n",
       "1   ./ncbi_genomes/JAAHGB01.fasta  ./query_genomes/JAAHGA01.fasta  0.000240   \n",
       "2   ./ncbi_genomes/JAAHGF01.fasta  ./query_genomes/JAAHGH01.fasta  0.041285   \n",
       "3   ./ncbi_genomes/JAAHHP01.fasta  ./query_genomes/JAAHHM01.fasta  0.000240   \n",
       "4     ./ncbi_genomes/AEPQ01.fasta  ./query_genomes/JAAHHD01.fasta  0.000781   \n",
       "5   ./ncbi_genomes/JAAHGW01.fasta  ./query_genomes/JAAHGU01.fasta  0.000192   \n",
       "6   ./ncbi_genomes/JAAHFR01.fasta  ./query_genomes/JAAHFU01.fasta  0.219531   \n",
       "7   ./ncbi_genomes/JAAHGA01.fasta  ./query_genomes/JAAHGB01.fasta  0.000240   \n",
       "8     ./ncbi_genomes/RCBZ01.fasta  ./query_genomes/JAAHFW01.fasta  0.004959   \n",
       "9   ./ncbi_genomes/JAAHGU01.fasta  ./query_genomes/JAAHGW01.fasta  0.000192   \n",
       "10  ./ncbi_genomes/JAAHGB01.fasta  ./query_genomes/JAAHGC01.fasta  0.000410   \n",
       "11    ./ncbi_genomes/ALVQ01.fasta  ./query_genomes/JAAHHN01.fasta  0.230110   \n",
       "12  ./ncbi_genomes/JAAHHC01.fasta  ./query_genomes/JAAHFZ01.fasta  0.016275   \n",
       "13  ./ncbi_genomes/JAAHGP01.fasta  ./query_genomes/JAAHGO01.fasta  0.051530   \n",
       "14  ./ncbi_genomes/JAAHHR01.fasta  ./query_genomes/JAAHGF01.fasta  0.013749   \n",
       "15  ./ncbi_genomes/JAAHFQ01.fasta  ./query_genomes/JAAHGS01.fasta  0.098138   \n",
       "16  ./ncbi_genomes/JAAHHD01.fasta  ./query_genomes/JAAHHC01.fasta  0.007596   \n",
       "17    ./ncbi_genomes/RFFC01.fasta  ./query_genomes/JAAHFN01.fasta  0.230110   \n",
       "18  ./ncbi_genomes/JAAHFQ01.fasta  ./query_genomes/JAAHHJ01.fasta  0.008967   \n",
       "19  ./ncbi_genomes/JAAHFO01.fasta  ./query_genomes/JAAHFP01.fasta  0.020280   \n",
       "20  ./ncbi_genomes/JAAHFM01.fasta  ./query_genomes/JAAHGL01.fasta  0.074432   \n",
       "21  ./ncbi_genomes/JAAHHQ01.fasta  ./query_genomes/JAAHIH01.fasta  0.024841   \n",
       "22  ./ncbi_genomes/JAAHGB01.fasta  ./query_genomes/JAAHGM01.fasta  0.034699   \n",
       "23    ./ncbi_genomes/QVFV01.fasta  ./query_genomes/JAAHII01.fasta  0.243761   \n",
       "\n",
       "           pvalue matching-hashes  \n",
       "0    0.000000e+00        990/1000  \n",
       "1    0.000000e+00        990/1000  \n",
       "2    0.000000e+00        266/1000  \n",
       "3    0.000000e+00        990/1000  \n",
       "4    0.000000e+00        968/1000  \n",
       "5    0.000000e+00        992/1000  \n",
       "6    5.955950e-18          5/1000  \n",
       "7    0.000000e+00        990/1000  \n",
       "8    0.000000e+00        820/1000  \n",
       "9    0.000000e+00        992/1000  \n",
       "10   0.000000e+00        983/1000  \n",
       "11   8.777560e-15          4/1000  \n",
       "12   0.000000e+00        551/1000  \n",
       "13   0.000000e+00        204/1000  \n",
       "14   0.000000e+00        599/1000  \n",
       "15  1.157750e-306         68/1000  \n",
       "16   0.000000e+00        743/1000  \n",
       "17   2.063740e-14          4/1000  \n",
       "18   0.000000e+00        707/1000  \n",
       "19   0.000000e+00        485/1000  \n",
       "20   0.000000e+00        117/1000  \n",
       "21   0.000000e+00        422/1000  \n",
       "22   0.000000e+00        318/1000  \n",
       "23   7.617380e-11          3/1000  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mash_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reference</th>\n",
       "      <th>query</th>\n",
       "      <th>distance</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>matching-hashes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>./ncbi_genomes/JAAHGU01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGW01.fasta</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>992/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>./ncbi_genomes/JAAHGW01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGU01.fasta</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>992/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./ncbi_genomes/JAAHHM01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHP01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>./ncbi_genomes/JAAHGB01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGA01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>./ncbi_genomes/JAAHHP01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHM01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>./ncbi_genomes/JAAHGA01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGB01.fasta</td>\n",
       "      <td>0.000240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>990/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>./ncbi_genomes/JAAHGB01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGC01.fasta</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>983/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>./ncbi_genomes/AEPQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHD01.fasta</td>\n",
       "      <td>0.000781</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>968/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>./ncbi_genomes/RCBZ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFW01.fasta</td>\n",
       "      <td>0.004959</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>820/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>./ncbi_genomes/JAAHHD01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHC01.fasta</td>\n",
       "      <td>0.007596</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>743/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>./ncbi_genomes/JAAHFQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHJ01.fasta</td>\n",
       "      <td>0.008967</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>707/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>./ncbi_genomes/JAAHHR01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGF01.fasta</td>\n",
       "      <td>0.013749</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>599/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>./ncbi_genomes/JAAHHC01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFZ01.fasta</td>\n",
       "      <td>0.016275</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>551/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>./ncbi_genomes/JAAHFO01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFP01.fasta</td>\n",
       "      <td>0.020280</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>485/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>./ncbi_genomes/JAAHHQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHIH01.fasta</td>\n",
       "      <td>0.024841</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>422/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>./ncbi_genomes/JAAHGB01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGM01.fasta</td>\n",
       "      <td>0.034699</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>318/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>./ncbi_genomes/JAAHGF01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGH01.fasta</td>\n",
       "      <td>0.041285</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>266/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>./ncbi_genomes/JAAHGP01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGO01.fasta</td>\n",
       "      <td>0.051530</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>204/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>./ncbi_genomes/JAAHFM01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGL01.fasta</td>\n",
       "      <td>0.074432</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>117/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>./ncbi_genomes/JAAHFQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHGS01.fasta</td>\n",
       "      <td>0.098138</td>\n",
       "      <td>1.157750e-306</td>\n",
       "      <td>68/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>./ncbi_genomes/JAAHFR01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFU01.fasta</td>\n",
       "      <td>0.219531</td>\n",
       "      <td>5.955950e-18</td>\n",
       "      <td>5/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>./ncbi_genomes/ALVQ01.fasta</td>\n",
       "      <td>./query_genomes/JAAHHN01.fasta</td>\n",
       "      <td>0.230110</td>\n",
       "      <td>8.777560e-15</td>\n",
       "      <td>4/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>./ncbi_genomes/RFFC01.fasta</td>\n",
       "      <td>./query_genomes/JAAHFN01.fasta</td>\n",
       "      <td>0.230110</td>\n",
       "      <td>2.063740e-14</td>\n",
       "      <td>4/1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>./ncbi_genomes/QVFV01.fasta</td>\n",
       "      <td>./query_genomes/JAAHII01.fasta</td>\n",
       "      <td>0.243761</td>\n",
       "      <td>7.617380e-11</td>\n",
       "      <td>3/1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        reference                           query  distance  \\\n",
       "9   ./ncbi_genomes/JAAHGU01.fasta  ./query_genomes/JAAHGW01.fasta  0.000192   \n",
       "5   ./ncbi_genomes/JAAHGW01.fasta  ./query_genomes/JAAHGU01.fasta  0.000192   \n",
       "0   ./ncbi_genomes/JAAHHM01.fasta  ./query_genomes/JAAHHP01.fasta  0.000240   \n",
       "1   ./ncbi_genomes/JAAHGB01.fasta  ./query_genomes/JAAHGA01.fasta  0.000240   \n",
       "3   ./ncbi_genomes/JAAHHP01.fasta  ./query_genomes/JAAHHM01.fasta  0.000240   \n",
       "7   ./ncbi_genomes/JAAHGA01.fasta  ./query_genomes/JAAHGB01.fasta  0.000240   \n",
       "10  ./ncbi_genomes/JAAHGB01.fasta  ./query_genomes/JAAHGC01.fasta  0.000410   \n",
       "4     ./ncbi_genomes/AEPQ01.fasta  ./query_genomes/JAAHHD01.fasta  0.000781   \n",
       "8     ./ncbi_genomes/RCBZ01.fasta  ./query_genomes/JAAHFW01.fasta  0.004959   \n",
       "16  ./ncbi_genomes/JAAHHD01.fasta  ./query_genomes/JAAHHC01.fasta  0.007596   \n",
       "18  ./ncbi_genomes/JAAHFQ01.fasta  ./query_genomes/JAAHHJ01.fasta  0.008967   \n",
       "14  ./ncbi_genomes/JAAHHR01.fasta  ./query_genomes/JAAHGF01.fasta  0.013749   \n",
       "12  ./ncbi_genomes/JAAHHC01.fasta  ./query_genomes/JAAHFZ01.fasta  0.016275   \n",
       "19  ./ncbi_genomes/JAAHFO01.fasta  ./query_genomes/JAAHFP01.fasta  0.020280   \n",
       "21  ./ncbi_genomes/JAAHHQ01.fasta  ./query_genomes/JAAHIH01.fasta  0.024841   \n",
       "22  ./ncbi_genomes/JAAHGB01.fasta  ./query_genomes/JAAHGM01.fasta  0.034699   \n",
       "2   ./ncbi_genomes/JAAHGF01.fasta  ./query_genomes/JAAHGH01.fasta  0.041285   \n",
       "13  ./ncbi_genomes/JAAHGP01.fasta  ./query_genomes/JAAHGO01.fasta  0.051530   \n",
       "20  ./ncbi_genomes/JAAHFM01.fasta  ./query_genomes/JAAHGL01.fasta  0.074432   \n",
       "15  ./ncbi_genomes/JAAHFQ01.fasta  ./query_genomes/JAAHGS01.fasta  0.098138   \n",
       "6   ./ncbi_genomes/JAAHFR01.fasta  ./query_genomes/JAAHFU01.fasta  0.219531   \n",
       "11    ./ncbi_genomes/ALVQ01.fasta  ./query_genomes/JAAHHN01.fasta  0.230110   \n",
       "17    ./ncbi_genomes/RFFC01.fasta  ./query_genomes/JAAHFN01.fasta  0.230110   \n",
       "23    ./ncbi_genomes/QVFV01.fasta  ./query_genomes/JAAHII01.fasta  0.243761   \n",
       "\n",
       "           pvalue matching-hashes  \n",
       "9    0.000000e+00        992/1000  \n",
       "5    0.000000e+00        992/1000  \n",
       "0    0.000000e+00        990/1000  \n",
       "1    0.000000e+00        990/1000  \n",
       "3    0.000000e+00        990/1000  \n",
       "7    0.000000e+00        990/1000  \n",
       "10   0.000000e+00        983/1000  \n",
       "4    0.000000e+00        968/1000  \n",
       "8    0.000000e+00        820/1000  \n",
       "16   0.000000e+00        743/1000  \n",
       "18   0.000000e+00        707/1000  \n",
       "14   0.000000e+00        599/1000  \n",
       "12   0.000000e+00        551/1000  \n",
       "19   0.000000e+00        485/1000  \n",
       "21   0.000000e+00        422/1000  \n",
       "22   0.000000e+00        318/1000  \n",
       "2    0.000000e+00        266/1000  \n",
       "13   0.000000e+00        204/1000  \n",
       "20   0.000000e+00        117/1000  \n",
       "15  1.157750e-306         68/1000  \n",
       "6    5.955950e-18          5/1000  \n",
       "11   8.777560e-15          4/1000  \n",
       "17   2.063740e-14          4/1000  \n",
       "23   7.617380e-11          3/1000  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mash_df.sort_values(by='distance')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Cateloging NCBI Genomes into Subsections**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from cyanotools1.miscellaneous import split_string_at\n",
    "\n",
    "def extract_ncbi_taxa(inputs_folder):\n",
    "    input_list = glob.glob(\"%s*.fasta\"%inputs_folder)\n",
    "    taxa = {}\n",
    "    for item in input_list:\n",
    "        records = list(SeqIO.parse(item, \"fasta\"))\n",
    "        strain = os.path.splitext(os.path.basename(item))[0]\n",
    "        if records:\n",
    "            if \"TPA_asm\" in str(records[0]):\n",
    "                genera = split_string_at(genera,\":\",1)[0]\n",
    "            else:\n",
    "                genera = split_string_at(records[0].description,\" \",1)[1]\n",
    "            if \"Candidatus\" in str(genera):\n",
    "                taxa[strain] = str(genera).partition(' ')[2].partition(' ')[0]\n",
    "            else:\n",
    "                taxa[strain] = str(genera).partition(' ')[0].lstrip(\"[\").rstrip(\"]\")\n",
    "    return taxa\n",
    "            \n",
    "taxa_dict = extract_ncbi_taxa(\"./refseq_cyano/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AADV02': 'Crocosphaera',\n",
       " 'AANO01': 'Synechococcus',\n",
       " 'AANP01': 'Synechococcus',\n",
       " 'AAOK01': 'Synechococcus',\n",
       " 'AATZ01': 'Synechococcus',\n",
       " 'AAUA01': 'Synechococcus',\n",
       " 'AAVU01': 'Lyngbya',\n",
       " 'AAVW01': 'Nodularia',\n",
       " 'AAXW01': 'Cyanothece',\n",
       " 'AAZV01': 'Leptolyngbya',\n",
       " 'ABRS01': 'Coleofasciculus',\n",
       " 'ABRV01': 'Synechococcus',\n",
       " 'ABSE01': 'Cyanobium',\n",
       " 'ABYK01': 'Arthrospira',\n",
       " 'ACDW01': 'Prochlorococcus',\n",
       " 'ACSK03': 'Arthrospira',\n",
       " 'ACYA01': 'Cylindrospermopsis',\n",
       " 'ACYB01': 'Raphidiopsis',\n",
       " 'ADXL01': 'Synechococcus',\n",
       " 'ADXM01': 'Synechococcus',\n",
       " 'AEPQ01': 'Moorea',\n",
       " 'AESD01': 'Crocosphaera',\n",
       " 'AFEJ01': 'Acaryochloris',\n",
       " 'AFJC01': 'Microcoleus',\n",
       " 'AFXD01': 'Arthrospira',\n",
       " 'AGCR01': 'Tolypothrix',\n",
       " 'AGIK01': 'Synechococcus',\n",
       " 'AGIZ01': 'Fischerella',\n",
       " 'AGJC02': 'Cyanothece',\n",
       " 'AHGV01': 'Thermanaerovibrio',\n",
       " 'AJHB01': 'Microcystis',\n",
       " 'AJLJ01': 'Fischerella',\n",
       " 'AJLK01': 'Fischerella',\n",
       " 'AJLL01': 'Fischerella',\n",
       " 'AJLM01': 'Chlorogloeopsis',\n",
       " 'AJLN01': 'Chlorogloeopsis',\n",
       " 'AJTX02': 'Prochlorothrix',\n",
       " 'AJUB01': 'Oscillatoriales',\n",
       " 'AJUS01': 'Microcystis',\n",
       " 'AJUT01': 'Microcystis',\n",
       " 'AJUU01': 'Microcystis',\n",
       " 'AJWF01': 'Anabaena',\n",
       " 'AJYB01': 'Planococcus',\n",
       " 'ALPB01': 'Prochlorococcus',\n",
       " 'ALPC01': 'Prochlorococcus',\n",
       " 'ALPD01': 'Prochlorococcus',\n",
       " 'ALPE01': 'Prochlorococcus',\n",
       " 'ALPF01': 'Prochlorococcus',\n",
       " 'ALPG01': 'Prochlorococcus',\n",
       " 'ALPH01': 'Prochlorococcus',\n",
       " 'ALPI01': 'Prochlorococcus',\n",
       " 'ALPJ01': 'Prochlorococcus',\n",
       " 'ALPK01': 'Prochlorococcus',\n",
       " 'ALPL01': 'Prochlorococcus',\n",
       " 'ALVI01': 'Kamptonema',\n",
       " 'ALVJ01': 'Calothrix',\n",
       " 'ALVK01': 'Pseudanabaena',\n",
       " 'ALVL01': 'Pleurocapsa',\n",
       " 'ALVM01': 'Leptolyngbya',\n",
       " 'ALVN01': 'Leptolyngbya',\n",
       " 'ALVO01': 'Geminocystis',\n",
       " 'ALVP01': 'Nodosilinea',\n",
       " 'ALVQ01': 'Spirulina',\n",
       " 'ALVR01': 'Spirulina',\n",
       " 'ALVS01': 'Fischerella',\n",
       " 'ALVT01': 'Fischerella',\n",
       " 'ALVU02': 'Synechocystis',\n",
       " 'ALVV02': 'Leptolyngbya',\n",
       " 'ALVX01': 'Fischerella',\n",
       " 'ALVY01': 'Gloeocapsa',\n",
       " 'ALVZ01': 'Xenococcus',\n",
       " 'ALWA01': 'Cyanobacterium',\n",
       " 'ALWB01': 'Pseudanabaena',\n",
       " 'ALWC01': 'Synechococcus',\n",
       " 'ALWD01': 'Scytonema',\n",
       " 'ANFJ01': 'Fortiea',\n",
       " 'ANFQ01': 'Geitlerinema',\n",
       " 'ANKN01': 'Prochlorothrix',\n",
       " 'ANKO01': 'Oscillatoria',\n",
       " 'ANKQ01': 'Microcystis',\n",
       " 'ANNX02': 'Scytonema',\n",
       " 'AOCI01': 'Microcystis',\n",
       " 'APIY01': 'Dolichospermum',\n",
       " 'APIZ01': 'Dolichospermum',\n",
       " 'AQPY01': 'Microcystis',\n",
       " 'ARCP01': 'Filamentous',\n",
       " 'ASAK01': 'Planktothrix',\n",
       " 'ASSJ01': 'Rubidibacter',\n",
       " 'ASZQ01': 'Microcystis',\n",
       " 'AUMZ01': 'Synechococcus',\n",
       " 'AUNA01': 'Synechococcus',\n",
       " 'AUSW01': 'Psychrobacter',\n",
       " 'AUZM01': 'Lyngbya',\n",
       " 'AVFS01': 'Planktothrix',\n",
       " 'AVFT01': 'Planktothrix',\n",
       " 'AVFU01': 'Planktothrix',\n",
       " 'AVFV01': 'Planktothrix',\n",
       " 'AVFW01': 'Planktothrix',\n",
       " 'AVFX01': 'Planktothrix',\n",
       " 'AVFY01': 'Planktothrix',\n",
       " 'AVFZ01': 'Planktothrix',\n",
       " 'AWNH01': 'Leptolyngbya',\n",
       " 'AXAQ01': 'Mastigocoleus',\n",
       " 'AXUV01': 'Synechococcus',\n",
       " 'AXUW01': 'Synechococcus',\n",
       " 'AXUX01': 'Synechococcus',\n",
       " 'AXZO01': 'Synechococcus',\n",
       " 'AZXL01': 'Synechococcus',\n",
       " 'AZYY01': 'Aphanizomenon',\n",
       " 'BAUB01': 'Synechococcus',\n",
       " 'BAWS01': 'Synechococcus',\n",
       " 'BBPA01': 'Microcystis',\n",
       " 'BBWW01': 'Leptolyngbya',\n",
       " 'BDGE01': 'Nostoc',\n",
       " 'BDSM01': 'Microcystis',\n",
       " 'BDUB01': 'Nodularia',\n",
       " 'BDUC01': 'Tolypothrix',\n",
       " 'BEIU01': 'Microcystis',\n",
       " 'BEYQ01': 'Microcystis',\n",
       " 'BFAC01': 'Microcystis',\n",
       " 'CACA01': 'Oscillatoria',\n",
       " 'CAFN01': 'Arthrospira',\n",
       " 'CAIH01': 'Microcystis',\n",
       " 'CAII01': 'Microcystis',\n",
       " 'CAIJ01': 'Microcystis',\n",
       " 'CAIK01': 'Microcystis',\n",
       " 'CAIL01': 'Microcystis',\n",
       " 'CAIM01': 'Microcystis',\n",
       " 'CAIN01': 'Microcystis',\n",
       " 'CAIO01': 'Microcystis',\n",
       " 'CAIP01': 'Microcystis',\n",
       " 'CAIQ01': 'Microcystis',\n",
       " 'CAIS01': 'Richelia',\n",
       " 'CAIY01': 'Richelia',\n",
       " 'CAQK01': 'Crocosphaera',\n",
       " 'CAQL01': 'Crocosphaera',\n",
       " 'CAQM01': 'Crocosphaera',\n",
       " 'CAQN01': 'Crocosphaera',\n",
       " 'CBZS01': 'Richelia',\n",
       " 'CDHJ01': 'Aphanizomenon',\n",
       " 'CVSV01': 'Prochlorococcus',\n",
       " 'CVSW01': 'Prochlorococcus',\n",
       " 'CVSX01': 'Prochlorococcus',\n",
       " 'CVSY01': 'Prochlorococcus',\n",
       " 'CVSZ01': 'Prochlorococcus',\n",
       " 'CZCS01': 'Planktothrix',\n",
       " 'CZCT02': 'Planktothrix',\n",
       " 'CZCU01': 'Planktothrix',\n",
       " 'CZCZ01': 'Planktothrix',\n",
       " 'CZDF01': 'Planktothrix',\n",
       " 'DAZV01': 'Prochlorococcus',\n",
       " 'DAZW01': 'Anabaena',\n",
       " 'DAZX01': 'Planktothrix',\n",
       " 'DBAX01': 'Prochlorococcus',\n",
       " 'DBCX01': 'Rhodoferax',\n",
       " 'DBIX01': 'Leptolyngbya',\n",
       " 'DBVD01': 'Cyanobium',\n",
       " 'DBVH01': 'Prochlorococcus',\n",
       " 'DBXD01': 'Microcystis',\n",
       " 'DBYM01': 'Prochlorococcus',\n",
       " 'DCNZ01': 'Anabaena',\n",
       " 'DCQC01': 'Aphanizomenon',\n",
       " 'DCSN01': 'Cyanobacteria',\n",
       " 'DDLS01': 'Anabaena',\n",
       " 'DDVE01': 'Cyanothece',\n",
       " 'DEDY01': 'Nostoc',\n",
       " 'DEEH01': 'Prochlorococcus',\n",
       " 'DEEN01': 'Phormidium',\n",
       " 'DEFJ01': 'Cyanobacteria',\n",
       " 'DEGT01': 'Microcystis',\n",
       " 'DEGZ01': 'Aphanizomenon',\n",
       " 'DESS01': 'Chroococcidiopsis',\n",
       " 'DESZ01': 'Nostoc',\n",
       " 'DFEL01': 'Prochlorococcus',\n",
       " 'DFQU01': 'Tolypothrix',\n",
       " 'DFRI01': 'Cyanobium',\n",
       " 'DFVO01': 'Richelia',\n",
       " 'DFYB01': 'Synechococcus',\n",
       " 'DGBR01': 'Aphanizomenon',\n",
       " 'DGDG01': 'Prochlorococcus',\n",
       " 'DGDH01': 'Prochlorococcus',\n",
       " 'DGYA01': 'Fischerella',\n",
       " 'DHMN01': 'Fischerella',\n",
       " 'DIYQ01': 'Mastigocoleus',\n",
       " 'DIYT01': 'Prochlorococcus',\n",
       " 'DJOE01': 'Crocosphaera',\n",
       " 'DJQF01': 'Prochlorococcus',\n",
       " 'DJYG01': 'Fischerella',\n",
       " 'DJYZ01': 'Cylindrospermopsis',\n",
       " 'DKMR01': 'Cyanobacteria',\n",
       " 'DKOB01': 'Cyanobium',\n",
       " 'DKOE01': 'Synechococcus',\n",
       " 'DKRT01': 'Prochlorococcus',\n",
       " 'DLDG01': 'Thermanaerovibrio',\n",
       " 'DLDQ01': 'Trichormus',\n",
       " 'DLEF01': 'Phormidium',\n",
       " 'DLEZ01': 'Synechocystis',\n",
       " 'DLGM01': 'Planktothrix',\n",
       " 'DLLM01': 'Prochlorococcus',\n",
       " 'DLPW01': 'Prochlorococcus',\n",
       " 'DLSU01': 'Filamentous',\n",
       " 'FITM01': 'Synechococcus',\n",
       " 'FSSI02': 'Oscillatoria',\n",
       " 'FXBO01': 'Synechococcus',\n",
       " 'FXWN01': 'Synechococcus',\n",
       " 'FYBH01': 'Calothrix',\n",
       " 'JENA01': 'Synechococcus',\n",
       " 'JFKN01': 'Prochlorococcus',\n",
       " 'JFKO01': 'Prochlorococcus',\n",
       " 'JFKP01': 'Prochlorococcus',\n",
       " 'JFKQ01': 'Prochlorococcus',\n",
       " 'JFKR01': 'Prochlorococcus',\n",
       " 'JFKS01': 'Prochlorococcus',\n",
       " 'JFKT01': 'Prochlorococcus',\n",
       " 'JFKU01': 'Prochlorococcus',\n",
       " 'JFKV01': 'Prochlorococcus',\n",
       " 'JFKW01': 'Prochlorococcus',\n",
       " 'JFKX01': 'Prochlorococcus',\n",
       " 'JFKY01': 'Prochlorococcus',\n",
       " 'JFKZ01': 'Prochlorococcus',\n",
       " 'JFLA01': 'Prochlorococcus',\n",
       " 'JFLB01': 'Prochlorococcus',\n",
       " 'JFLC01': 'Prochlorococcus',\n",
       " 'JFLD01': 'Prochlorococcus',\n",
       " 'JFLE01': 'Prochlorococcus',\n",
       " 'JFLF01': 'Prochlorococcus',\n",
       " 'JFLG01': 'Prochlorococcus',\n",
       " 'JFLH01': 'Prochlorococcus',\n",
       " 'JFLI01': 'Prochlorococcus',\n",
       " 'JFLJ01': 'Prochlorococcus',\n",
       " 'JFLK01': 'Prochlorococcus',\n",
       " 'JFLL01': 'Prochlorococcus',\n",
       " 'JFLM01': 'Prochlorococcus',\n",
       " 'JFLN01': 'Prochlorococcus',\n",
       " 'JFLO01': 'Prochlorococcus',\n",
       " 'JFLP01': 'Prochlorococcus',\n",
       " 'JFLQ01': 'Prochlorococcus',\n",
       " 'JFLR01': 'Prochlorococcus',\n",
       " 'JFLS01': 'Prochlorococcus',\n",
       " 'JFLT01': 'Prochlorococcus',\n",
       " 'JFLU01': 'Prochlorococcus',\n",
       " 'JFLV01': 'Prochlorococcus',\n",
       " 'JFLW01': 'Prochlorococcus',\n",
       " 'JFLX01': 'Prochlorococcus',\n",
       " 'JFLY01': 'Prochlorococcus',\n",
       " 'JFLZ01': 'Prochlorococcus',\n",
       " 'JFMA01': 'Prochlorococcus',\n",
       " 'JFMB01': 'Prochlorococcus',\n",
       " 'JFMC01': 'Prochlorococcus',\n",
       " 'JFMD01': 'Prochlorococcus',\n",
       " 'JFME01': 'Prochlorococcus',\n",
       " 'JFMF01': 'Prochlorococcus',\n",
       " 'JFMG01': 'Prochlorococcus',\n",
       " 'JFMH01': 'Prochlorococcus',\n",
       " 'JFMI01': 'Prochlorococcus',\n",
       " 'JFMJ01': 'Prochlorococcus',\n",
       " 'JFMK01': 'Prochlorococcus',\n",
       " 'JFML01': 'Prochlorococcus',\n",
       " 'JFMM01': 'Prochlorococcus',\n",
       " 'JFMN01': 'Prochlorococcus',\n",
       " 'JFMO01': 'Prochlorococcus',\n",
       " 'JFMP01': 'Prochlorococcus',\n",
       " 'JFMQ01': 'Prochlorococcus',\n",
       " 'JFMR01': 'Prochlorococcus',\n",
       " 'JFMS01': 'Prochlorococcus',\n",
       " 'JFMT01': 'Prochlorococcus',\n",
       " 'JFMU01': 'Prochlorococcus',\n",
       " 'JFMV01': 'Prochlorococcus',\n",
       " 'JFMW01': 'Prochlorococcus',\n",
       " 'JFMX01': 'Prochlorococcus',\n",
       " 'JFMY01': 'Prochlorococcus',\n",
       " 'JFMZ01': 'Prochlorococcus',\n",
       " 'JFNA01': 'Prochlorococcus',\n",
       " 'JFNB01': 'Prochlorococcus',\n",
       " 'JFNC01': 'Prochlorococcus',\n",
       " 'JFND01': 'Prochlorococcus',\n",
       " 'JFNE01': 'Prochlorococcus',\n",
       " 'JFNF01': 'Prochlorococcus',\n",
       " 'JFNG01': 'Prochlorococcus',\n",
       " 'JFNH01': 'Prochlorococcus',\n",
       " 'JFNI01': 'Prochlorococcus',\n",
       " 'JFNJ01': 'Prochlorococcus',\n",
       " 'JFNK01': 'Prochlorococcus',\n",
       " 'JFNL01': 'Prochlorococcus',\n",
       " 'JFNM01': 'Prochlorococcus',\n",
       " 'JFNN01': 'Prochlorococcus',\n",
       " 'JFNO01': 'Prochlorococcus',\n",
       " 'JFNP01': 'Prochlorococcus',\n",
       " 'JFNQ01': 'Prochlorococcus',\n",
       " 'JFNR01': 'Prochlorococcus',\n",
       " 'JFNS01': 'Prochlorococcus',\n",
       " 'JFNT01': 'Prochlorococcus',\n",
       " 'JFNU01': 'Prochlorococcus',\n",
       " 'JFNV01': 'Prochlorococcus',\n",
       " 'JFNW01': 'Prochlorococcus',\n",
       " 'JFNX01': 'Prochlorococcus',\n",
       " 'JFNY01': 'Prochlorococcus',\n",
       " 'JFNZ01': 'Prochlorococcus',\n",
       " 'JFOA01': 'Prochlorococcus',\n",
       " 'JFOB01': 'Prochlorococcus',\n",
       " 'JFOC01': 'Prochlorococcus',\n",
       " 'JFOD01': 'Prochlorococcus',\n",
       " 'JFOE01': 'Prochlorococcus',\n",
       " 'JHEG02': 'Tolypothrix',\n",
       " 'JJML01': 'Neosynechococcus',\n",
       " 'JMIX01': 'Erythrobacter',\n",
       " 'JMKF01': 'Leptolyngbya',\n",
       " 'JMRP01': 'Cyanobium',\n",
       " 'JNAG01': 'Prochlorococcus',\n",
       " 'JNAH01': 'Prochlorococcus',\n",
       " 'JNAI01': 'Prochlorococcus',\n",
       " 'JNAJ01': 'Prochlorococcus',\n",
       " 'JNAK01': 'Prochlorococcus',\n",
       " 'JNAL01': 'Prochlorococcus',\n",
       " 'JNAM01': 'Prochlorococcus',\n",
       " 'JNAN01': 'Prochlorococcus',\n",
       " 'JNAO01': 'Prochlorococcus',\n",
       " 'JNAP01': 'Prochlorococcus',\n",
       " 'JNAQ01': 'Prochlorococcus',\n",
       " 'JNAR01': 'Prochlorococcus',\n",
       " 'JNAS01': 'Prochlorococcus',\n",
       " 'JNAT01': 'Prochlorococcus',\n",
       " 'JNAU01': 'Prochlorococcus',\n",
       " 'JNAV01': 'Prochlorococcus',\n",
       " 'JNAW01': 'Prochlorococcus',\n",
       " 'JNAX01': 'Prochlorococcus',\n",
       " 'JNAY01': 'Prochlorococcus',\n",
       " 'JNAZ01': 'Prochlorococcus',\n",
       " 'JNBA01': 'Prochlorococcus',\n",
       " 'JNBB01': 'Prochlorococcus',\n",
       " 'JNBC01': 'Prochlorococcus',\n",
       " 'JNBD01': 'Prochlorococcus',\n",
       " 'JNBE01': 'Prochlorococcus',\n",
       " 'JPSP01': 'Atelocyanobacterium',\n",
       " 'JQFA01': 'Leptolyngbya',\n",
       " 'JRFE01': 'Myxosarcina',\n",
       " 'JSDP01': 'Aphanizomenon',\n",
       " 'JSUH01': 'Kocuria',\n",
       " 'JTCM01': 'Hassallia',\n",
       " 'JTHE01': 'Lyngbya',\n",
       " 'JTJC02': 'Scytonema',\n",
       " 'JTJD01': 'Aphanocapsa',\n",
       " 'JXCA01': 'Scytonema',\n",
       " 'JXCB01': 'Tolypothrix',\n",
       " 'JXIJ01': 'Mastigocladus',\n",
       " 'JXQG01': 'Synechococcus',\n",
       " 'JXUO01': 'Synechococcus',\n",
       " 'JXYX01': 'Microcystis',\n",
       " 'JYFQ01': 'Synechococcus',\n",
       " 'JYON01': 'Aliterella',\n",
       " 'LADK01': 'Trichodesmium',\n",
       " 'LAGX01': 'Nostoc',\n",
       " 'LAGY01': 'Nostoc',\n",
       " 'LAGZ01': 'Nostoc',\n",
       " 'LAHA01': 'Nostoc',\n",
       " 'LAHB01': 'Nostoc',\n",
       " 'LAHC01': 'Nostoc',\n",
       " 'LAHD01': 'Nostoc',\n",
       " 'LAHE01': 'Nostoc',\n",
       " 'LAHF01': 'Nostoc',\n",
       " 'LAHG01': 'Nostoc',\n",
       " 'LAHH01': 'Nostoc',\n",
       " 'LAHI01': 'Nostoc',\n",
       " 'LAHJ01': 'Nostoc',\n",
       " 'LAMW01': 'Trichodesmium',\n",
       " 'LATL02': 'Limnoraphis',\n",
       " 'LAYT01': 'Arthrospira',\n",
       " 'LFEK01': 'Synechococcus',\n",
       " 'LGSU01': 'Hydrocoleum',\n",
       " 'LICO01': 'Cyanobacterium',\n",
       " 'LIRE01': 'Pseudanabaena',\n",
       " 'LIRN01': 'Hapalosiphon',\n",
       " 'LIRO01': 'Limnothrix',\n",
       " 'LIUQ01': 'Planktothricoides',\n",
       " 'LJOP01': 'Anabaena',\n",
       " 'LJOQ01': 'Anabaena',\n",
       " 'LJOS01': 'Anabaena',\n",
       " 'LJOT01': 'Anabaena',\n",
       " 'LJOU01': 'Anabaena',\n",
       " 'LJOV01': 'Anabaena',\n",
       " 'LJOW01': 'Aphanizomenon',\n",
       " 'LJOX01': 'Aphanizomenon',\n",
       " 'LJOY01': 'Aphanizomenon',\n",
       " 'LJZR01': 'Phormidesmis',\n",
       " 'LJZT01': 'Phormidium',\n",
       " 'LMTZ01': 'Mastigocoleus',\n",
       " 'LMVE01': 'Cylindrospermopsis',\n",
       " 'LNAA01': 'Oscillatoriales',\n",
       " 'LNDC01': 'Geitlerinema',\n",
       " 'LSSA01': 'Nostoc',\n",
       " 'LSYZ01': 'Leptolyngbya',\n",
       " 'LSZA01': 'Phormidium',\n",
       " 'LTEC01': 'Aphanizomenon',\n",
       " 'LUBZ01': 'Cylindrospermopsis',\n",
       " 'LUFH01': 'Sphaerospermopsis',\n",
       " 'LUGL01': 'Elainella',\n",
       " 'LUHI01': 'Anabaena',\n",
       " 'LUHJ01': 'Anabaena',\n",
       " 'LUPS01': 'Prochlorococcus',\n",
       " 'LUPT01': 'Prochlorococcus',\n",
       " 'LUPU01': 'Prochlorococcus',\n",
       " 'LUPV01': 'Prochlorococcus',\n",
       " 'LURJ01': 'Synechococcus',\n",
       " 'LURK01': 'Synechococcus',\n",
       " 'LVHK01': 'Prochlorococcus',\n",
       " 'LVHL01': 'Prochlorococcus',\n",
       " 'LVHM01': 'Prochlorococcus',\n",
       " 'LVHN01': 'Prochlorococcus',\n",
       " 'LVHO01': 'Prochlorococcus',\n",
       " 'LVHP01': 'Prochlorococcus',\n",
       " 'LVHQ01': 'Prochlorococcus',\n",
       " 'LVHR01': 'Prochlorococcus',\n",
       " 'LVHS01': 'Prochlorococcus',\n",
       " 'LVHT01': 'Synechococcus',\n",
       " 'LVHU01': 'Synechococcus',\n",
       " 'LVHV01': 'Synechococcus',\n",
       " 'LWAJ01': 'Nodularia',\n",
       " 'LWHC01': 'Cyanobacterium',\n",
       " 'LWRO01': 'Geitlerinema',\n",
       " 'LXYR01': 'Phormidesmis',\n",
       " 'LYXA01': 'Cylindrospermopsis',\n",
       " 'LZQD01': 'Trichormus',\n",
       " 'MBQX01': 'Cylindrospermopsis',\n",
       " 'MBQY01': 'Cylindrospermopsis',\n",
       " 'MBRD01': 'Nostoc',\n",
       " 'MBRE01': 'Oscillatoriales',\n",
       " 'MBRF01': 'Limnothrix',\n",
       " 'MCIH01': 'Microcystis',\n",
       " 'MDVL01': 'Halothece',\n",
       " 'MDZH01': 'Microcystis',\n",
       " 'MJGC01': 'Desertifilum',\n",
       " 'MKGP01': 'Limnothrix',\n",
       " 'MKZR01': 'Moorea',\n",
       " 'MKZS01': 'Moorea',\n",
       " 'MLAW01': 'Roseofilum',\n",
       " 'MNHI01': 'Cyanobacteria',\n",
       " 'MNIV01': 'Cyanobacteria',\n",
       " 'MNPM01': 'Mastigocladus',\n",
       " 'MNYZ01': 'Oscillatoriales',\n",
       " 'MNZA01': 'Oscillatoriales',\n",
       " 'MOLN01': 'Microcystis',\n",
       " 'MOLZ01': 'Microcystis',\n",
       " 'MPMK01': 'Cyanobacteria',\n",
       " 'MPPI01': 'Phormidesmis',\n",
       " 'MQTZ01': 'Leptolyngbya',\n",
       " 'MRBY01': 'Limnothrix',\n",
       " 'MRBZ01': 'Nostoc',\n",
       " 'MRCA01': 'Fischerella',\n",
       " 'MRCB01': 'Hydrococcus',\n",
       " 'MRCC01': 'Chroogloeocystis',\n",
       " 'MRCD01': 'Calothrix',\n",
       " 'MRCE01': 'Phormidium',\n",
       " 'MRCF01': 'Scytonema',\n",
       " 'MRCG01': 'Phormidium',\n",
       " 'MRUD01': 'Synechococcus',\n",
       " 'MTAV01': 'Nostoc',\n",
       " 'MTAW01': 'Nostoc',\n",
       " 'MTAX01': 'Nostoc',\n",
       " 'MTBS01': 'Microcystis',\n",
       " 'MTBT01': 'Microcystis',\n",
       " 'MTBU01': 'Microcystis',\n",
       " 'MTPU01': 'Cylindrospermopsis',\n",
       " 'MTQM01': 'Microbacterium',\n",
       " 'MUGG01': 'Alkalinema',\n",
       " 'MVDI01': 'Chroococcidiopsis',\n",
       " 'MVGR01': 'Microcystis',\n",
       " 'MVOP01': 'Prochlorococcus',\n",
       " 'MWLD01': 'Synechococcus',\n",
       " 'MWLE01': 'Synechococcus',\n",
       " 'MWLF01': 'Synechococcus',\n",
       " 'MWLG01': 'Synechococcus',\n",
       " 'MWMJ01': 'Synechococcus',\n",
       " 'MWOO01': 'Prochlorococcus',\n",
       " 'MWOP01': 'Prochlorococcus',\n",
       " 'MWOQ01': 'Prochlorococcus',\n",
       " 'MWOR01': 'Prochlorococcus',\n",
       " 'MWOS01': 'Prochlorococcus',\n",
       " 'MWOT01': 'Prochlorococcus',\n",
       " 'MWOU01': 'Prochlorococcus',\n",
       " 'MWOV01': 'Prochlorococcus',\n",
       " 'MWOW01': 'Prochlorococcus',\n",
       " 'MWOX01': 'Prochlorococcus',\n",
       " 'MWOY01': 'Prochlorococcus',\n",
       " 'MWOZ01': 'Prochlorococcus',\n",
       " 'MWPA01': 'Prochlorococcus',\n",
       " 'MWPC01': 'Prochlorococcus',\n",
       " 'MWPD01': 'Prochlorococcus',\n",
       " 'MWPE01': 'Prochlorococcus',\n",
       " 'MWPF01': 'Prochlorococcus',\n",
       " 'MWPG01': 'Prochlorococcus',\n",
       " 'NBYN01': 'Cylindrospermopsis',\n",
       " 'NDGV01': 'Cyanobacterium',\n",
       " 'NDHW01': 'Pseudanabaena',\n",
       " 'NEED01': 'Unicellular',\n",
       " 'NETF01': 'Unicellular',\n",
       " 'NGZS01': 'Fischerella',\n",
       " 'NGZT01': 'Fischerella',\n",
       " 'NGZU01': 'Fischerella',\n",
       " 'NGZV01': 'Fischerella',\n",
       " 'NGZW01': 'Fischerella',\n",
       " 'NGZX01': 'Fischerella',\n",
       " 'NGZY01': 'Fischerella',\n",
       " 'NGZZ01': 'Fischerella',\n",
       " 'NHAA01': 'Fischerella',\n",
       " 'NHAB01': 'Fischerella',\n",
       " 'NHAC01': 'Fischerella',\n",
       " 'NHAD01': 'Fischerella',\n",
       " 'NHAE01': 'Fischerella',\n",
       " 'NHAF01': 'Fischerella',\n",
       " 'NHAG01': 'Fischerella',\n",
       " 'NHAH01': 'Fischerella',\n",
       " 'NHAI01': 'Fischerella',\n",
       " 'NHAJ01': 'Fischerella',\n",
       " 'NHAK01': 'Fischerella',\n",
       " 'NHBY01': 'Synechococcus',\n",
       " 'NHBZ01': 'Synechococcus',\n",
       " 'NHDT01': 'Synechococcus',\n",
       " 'NHER01': 'Synechococcus',\n",
       " 'NHHE01': 'Synechococcus',\n",
       " 'NHHS01': 'Synechococcus',\n",
       " 'NHIA01': 'Cyanobacteria',\n",
       " 'NHII01': 'Synechococcus',\n",
       " 'NHIK01': 'Synechococcus',\n",
       " 'NHIL01': 'Cyanobacteria',\n",
       " 'NHJC01': 'Synechococcus',\n",
       " 'NHJU01': 'Prochlorococcus',\n",
       " 'NHKA01': 'Cyanobacteria',\n",
       " 'NJHS01': 'Cylindrospermopsis',\n",
       " 'NJHT01': 'Cylindrospermopsis',\n",
       " 'NJHU01': 'Cylindrospermopsis',\n",
       " 'NJHV01': 'Cylindrospermopsis',\n",
       " 'NJHW01': 'Cylindrospermopsis',\n",
       " 'NJHX01': 'Cylindrospermopsis',\n",
       " 'NJHY01': 'Cylindrospermopsis',\n",
       " 'NJHZ01': 'Cylindrospermopsis',\n",
       " 'NJIA01': 'Cylindrospermopsis',\n",
       " 'NKFP01': 'Leptolyngbya',\n",
       " 'NMQA01': 'Fischerella',\n",
       " 'NMQB01': 'Fischerella',\n",
       " 'NMQC01': 'Fischerella',\n",
       " 'NMQD01': 'Fischerella',\n",
       " 'NMQE01': 'Fischerella',\n",
       " 'NMQF01': 'Fischerella',\n",
       " 'NMQG01': 'Fischerella',\n",
       " 'NMQH01': 'Fischerella',\n",
       " 'NMQI01': 'Fischerella',\n",
       " 'NMQJ01': 'Fischerella',\n",
       " 'NMQK01': 'Fischerella',\n",
       " 'NOLI01': 'Nostoc',\n",
       " 'NOLJ01': 'Nostoc',\n",
       " 'NOLK01': 'Nostoc',\n",
       " 'NOXS01': 'Elstera',\n",
       " 'NOXT01': 'Sandarakinorhabdus',\n",
       " 'NOXU01': 'Niveispirillum',\n",
       " 'NOXV01': 'Flavobacterium',\n",
       " 'NOXW01': 'Rhodoferax',\n",
       " 'NOXX01': 'Flavobacterium',\n",
       " 'NPIK01': 'Cylindrospermopsis',\n",
       " 'NPIL01': 'Cylindrospermopsis',\n",
       " 'NQKW01': 'Synechococcus',\n",
       " 'NQKX01': 'Synechococcus',\n",
       " 'NQKY01': 'Synechococcus',\n",
       " 'NQKZ01': 'Synechococcus',\n",
       " 'NQLA01': 'Synechococcus',\n",
       " 'NQMD01': 'Fischerella',\n",
       " 'NQME01': 'Fischerella',\n",
       " 'NQMF01': 'Fischerella',\n",
       " 'NRIU01': 'Geitlerinema',\n",
       " 'NRQW01': 'Fischerella',\n",
       " 'NRTA01': 'Leptolyngbya',\n",
       " 'NSHF01': 'Nostoc',\n",
       " 'NSHT01': 'Cyanobium',\n",
       " 'NSII01': 'Synechococcus',\n",
       " 'NTFS01': 'Calothrix',\n",
       " 'NXIB02': 'Tychonema',\n",
       " 'NYUU01': 'Synechococcus',\n",
       " 'NYXI01': 'Synechococcus',\n",
       " 'NYZI01': 'Cyanobium',\n",
       " 'NZED01': 'Synechococcus',\n",
       " 'NZHI01': 'Synechococcus',\n",
       " 'NZHY01': 'Prochlorococcus',\n",
       " 'NZJI01': 'Cyanobium',\n",
       " 'NZKA01': 'Prochlorococcus',\n",
       " 'NZPI01': 'Synechococcus',\n",
       " 'NZVB01': 'Cyanobium',\n",
       " 'NZXF01': 'Synechococcus',\n",
       " 'PAEP01': 'Synechococcus',\n",
       " 'PAEU01': 'Cyanobium',\n",
       " 'PAJY01': 'Synechococcus',\n",
       " 'PAXG01': 'Cyanobium',\n",
       " 'PBHJ01': 'Prochlorococcus',\n",
       " 'PBPY01': 'Synechococcus',\n",
       " 'PBVK01': 'Cyanobium',\n",
       " 'PBYZ01': 'Synechococcus',\n",
       " 'PCGE01': 'Synechococcus',\n",
       " 'PDDM01': 'Pseudanabaena',\n",
       " 'PEBC01': 'Cyanobacterium',\n",
       " 'PEIG01': 'Chroococcales',\n",
       " 'PESY01': 'Prochlorococcus',\n",
       " 'PGEM01': 'Cuspidothrix',\n",
       " 'PHFO01': 'Cyanobacteria',\n",
       " 'PKDM01': 'Cyanobacterium',\n",
       " 'PKDN01': 'Cyanobacterium',\n",
       " 'PKRO01': 'Porphyrobacter',\n",
       " 'PKRP01': 'Rhizobium',\n",
       " 'PKRQ01': 'Tabrizicola',\n",
       " 'PKRR01': 'Uliginosibacterium',\n",
       " 'PKRS01': 'Emticicia',\n",
       " 'PKRT01': 'Novosphingobium',\n",
       " 'PRLH01': 'Cyanothece',\n",
       " 'PTJN01': 'Gloeocapsopsis',\n",
       " 'PVWD01': 'Filamentous',\n",
       " 'PVWE01': 'Filamentous',\n",
       " 'PVWF01': 'Pleurocapsa',\n",
       " 'PVWG01': 'Phormidesmis',\n",
       " 'PVWH01': 'Filamentous',\n",
       " 'PVWI01': 'Filamentous',\n",
       " 'PVWJ01': 'Merismopedia',\n",
       " 'PVWK01': 'Leptolyngbya',\n",
       " 'PVWL01': 'Cyanosarcina',\n",
       " 'PVWM01': 'Chroococcidiopsis',\n",
       " 'PVWN01': 'Chlorogloea',\n",
       " 'PVWO01': 'Chamaesiphon',\n",
       " 'PVWP01': 'Aphanothece',\n",
       " 'PXOH01': 'Aphanothece',\n",
       " 'PXPI01': 'Cyanobacteria',\n",
       " 'PXPJ01': 'Cyanobacteria',\n",
       " 'PXPK01': 'Cyanobacteria',\n",
       " 'PXPL01': 'Cyanobacteria',\n",
       " 'PXPM01': 'Cyanobacteria',\n",
       " 'PXPN01': 'Cyanobacteria',\n",
       " 'PXPO01': 'Cyanobacteria',\n",
       " 'PXPP01': 'Cyanobacteria',\n",
       " 'PXPQ01': 'Cyanobacteria',\n",
       " 'PXPR01': 'Cyanobacteria',\n",
       " 'PXPS01': 'Cyanobacteria',\n",
       " 'PXPT01': 'Cyanobacteria',\n",
       " 'PXPU01': 'Cyanobacteria',\n",
       " 'PXPV01': 'Cyanobacteria',\n",
       " 'PXPW01': 'Cyanobacteria',\n",
       " 'PXPX01': 'Cyanobacteria',\n",
       " 'PXPY01': 'Cyanobacteria',\n",
       " 'PXPZ01': 'Cyanobacteria',\n",
       " 'PXQA01': 'Cyanobacteria',\n",
       " 'PXQB01': 'Cyanobacteria',\n",
       " 'PXQC01': 'Cyanobacteria',\n",
       " 'PXQD01': 'Cyanobacteria',\n",
       " 'PXQE01': 'Cyanobacteria',\n",
       " 'PXQF01': 'Cyanobacteria',\n",
       " 'PXQG01': 'Cyanobacteria',\n",
       " 'PXQH01': 'Cyanobacteria',\n",
       " 'PXQI01': 'Cyanobacteria',\n",
       " 'PXVC01': 'Synechococcus',\n",
       " 'PXXO01': 'Cyanobium',\n",
       " 'PYCI01': 'Chroococcidiopsis',\n",
       " 'PYEQ01': 'Filamentous',\n",
       " 'PYER01': 'Filamentous',\n",
       " 'PYFY01': 'Filamentous',\n",
       " 'PYGV01': 'Filamentous'}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxa_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "\n",
    "with open('./mash-Linux64-v2.0/mash_taxonomy_1.csv', 'wb') as csv_file:\n",
    "    writer = csv.writer(csv_file)\n",
    "    for key, value in taxa_dict.items():\n",
    "       writer.writerow([key, value])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat ./mash-Linux64-v2.0/mash_taxonomy_1.csv >> ./mash-Linux64-v2.0/mash_taxonomy.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ADXL01,Synechococcus\r",
      "\r\n",
      "BFAC01,Microcystis\r",
      "\r\n",
      "PXQE01,Cyanobacteria\r",
      "\r\n",
      "JNAR01,Prochlorococcus\r",
      "\r\n",
      "ALPC01,Prochlorococcus\r",
      "\r\n",
      "DLEF01,Phormidium\r",
      "\r\n",
      "DEGZ01,Aphanizomenon\r",
      "\r\n",
      "AOCI01,Microcystis\r",
      "\r\n",
      "ASSJ01,Rubidibacter\r",
      "\r\n",
      "LADK01,Trichodesmium\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!tail ./mash-Linux64-v2.0/mash_taxonomy.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Using MASH Distances to Catelog Cyanobiome Finished Genomes into Subsections**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3H3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1I7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1A4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2B6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1I4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2D2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3A5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2C4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3C1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2C1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2C8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1E3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1G6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3B5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2B7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2B4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1A9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3E8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3A1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2E9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1G2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3C4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2B2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3E9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3C5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3B1.fasta > ./mash-Linux64-v2.0/temp.txt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1H6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1F2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1G1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4B2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2E6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1H7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1G5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1E4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3E1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1B2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4F7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3I5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4G3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2E2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1I9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2E3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2C9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2D1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2E7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1F9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3I6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3C2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3C8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3I7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2B9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I6.fasta > ./mash-Linux64-v2.0/temp.txt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3I9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2D8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1I2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1H4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2C2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3C6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3E2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1B9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1H2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1D4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4B1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1D8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2A8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2A9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2H1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3A2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3A4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1E1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2F1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3H2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1B1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3H8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3E6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4B8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3E5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3I8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2B3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1A7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2I8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3B2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4G2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4D6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3B8.fasta > ./mash-Linux64-v2.0/temp.txt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3A7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1A2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4F4.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3B3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3H1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G2.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1A3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4B7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4C9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3F3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1B8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2H7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4D8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2A1.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1C9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1B6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3G9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2C3.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1D6.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2G5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1H5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/1D9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4D9.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/2H8.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4D7.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "./mash-Linux64-v2.0/mash dist ./mash-Linux64-v2.0/ncbi_genomes.msh /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/3H5.fasta > ./mash-Linux64-v2.0/temp.txt\n",
      "\n",
      "--- 124.882939816 seconds ---\n"
     ]
    }
   ],
   "source": [
    "def get_mash_df(inputs_folder,sketch):\n",
    "    start_time = time.time()\n",
    "    input_list = glob.glob(\"%s*.fasta\"%inputs_folder)\n",
    "    final_df = pd.DataFrame()\n",
    "    for index,item in enumerate(input_list):\n",
    "        scores = []\n",
    "        cmd = \"./mash-Linux64-v2.0/mash dist %s %s > ./mash-Linux64-v2.0/temp.txt\"%(sketch,item)\n",
    "        subprocess.call(cmd,shell=True)\n",
    "        df = pd.read_csv(\"./mash-Linux64-v2.0/temp.txt\",sep=\"\\t\",names=[\"reference\",\"query\",\"distance\",\"pvalue\",\"matching-hashes\"])\n",
    "        df = df[df.reference != item]\n",
    "        row = df[df.distance == df.distance.min()]\n",
    "        final_df = final_df.append(row[:1], ignore_index = True)\n",
    "    print('\\n' + \"--- %s seconds ---\" %(time.time()-start_time))\n",
    "    return final_df\n",
    "        \n",
    "mash_df = get_mash_df(\"/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/\",\"./mash-Linux64-v2.0/ncbi_genomes.msh\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'mash_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-70563605d368>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mtaxa_dict2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mr\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmash_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miterrows\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0mncbi_taxa\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplitext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbasename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'.'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mstrain\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplitext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbasename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'_'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'mash_df' is not defined"
     ]
    }
   ],
   "source": [
    "taxa_dict2 = {}\n",
    "\n",
    "for i,r in mash_df.iterrows():\n",
    "    ncbi_taxa = os.path.splitext(os.path.basename((r[0])))[0].partition('.')[0]\n",
    "    strain = os.path.splitext(os.path.basename((r[1])))[0].partition('_')[0]\n",
    "    taxa_dict2[strain] = taxa_dict[ncbi_taxa]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'taxa_dict2' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-e9a615a51ffa>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtaxa_dict2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"3B5\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'taxa_dict2' is not defined"
     ]
    }
   ],
   "source": [
    "taxa_dict2[\"3B5\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "\n",
    "with open('./mash-Linux64-v2.0/mash_taxonomy.csv', 'wb') as csv_file:\n",
    "    writer = csv.writer(csv_file)\n",
    "    for key, value in taxa_dict2.items():\n",
    "       writer.writerow([key, value])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
