{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`2019-09-07` `Tiago Ferreira Leao` **v1.0**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cyanobiome master dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "Requirements: packages below\n",
    "              Installing CheckM\n",
    "              Assembled genomes (from metagenomes)(run processing_genome_assembly)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Loading packages**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'Bio'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-5885de9336df>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mglob\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeqUtils\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mGC\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'Bio'"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import GC\n",
    "import numpy as np\n",
    "import re\n",
    "import subprocess\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Running CheckM in bash**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "checkm taxonomy_wf phylum Cyanobacteria -x fasta /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/ /home/tiago/Desktop/cyanet/checkm/ >> /home/tiago/Desktop/cyanet/checkm/cyanobiome_checkm-TFL181108.tsv\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Loading CheckM dataframe for scaffolded genomes**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_loci = '/home/tiago/Desktop/cyanet/tables/genomic_loci_4-TFL190105.txt'\n",
    "scaffolds_path = \"/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/\"\n",
    "checkm_file = \"/home/tiago/Desktop/cyanet/checkm/cyanobiome_checkm-TFL181108.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_checkm_data(filename):\n",
    "    headers=['bin_id','marker_lineage','marker_number','genomes','markers','marker_sets','0','1','2','3','4','5+','completeness','contamination','strain_heterogeneity']\n",
    "    df = pd.read_csv(filename, delimiter=r'\\s+',names=headers,skiprows=3)\n",
    "    df = df.drop(df.index[len(df)-1])\n",
    "    return df\n",
    "\n",
    "checkm_df1 = get_checkm_data(checkm_file)\n",
    "checkm_dict = pd.Series(checkm_df1[\"completeness\"].values,index=checkm_df1[\"bin_id\"]).to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(161, 15)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkm_df1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_id</th>\n",
       "      <th>marker_lineage</th>\n",
       "      <th>marker_number</th>\n",
       "      <th>genomes</th>\n",
       "      <th>markers</th>\n",
       "      <th>marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3C6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>380</td>\n",
       "      <td>89</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>18.57</td>\n",
       "      <td>8.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2I9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>411</td>\n",
       "      <td>52</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>114.24</td>\n",
       "      <td>46.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2I6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>220</td>\n",
       "      <td>179</td>\n",
       "      <td>57</td>\n",
       "      <td>12</td>\n",
       "      <td>100.00</td>\n",
       "      <td>166.73</td>\n",
       "      <td>4.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2H1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>117</td>\n",
       "      <td>299</td>\n",
       "      <td>47</td>\n",
       "      <td>6</td>\n",
       "      <td>100.00</td>\n",
       "      <td>185.79</td>\n",
       "      <td>4.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2G7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>259</td>\n",
       "      <td>112</td>\n",
       "      <td>35</td>\n",
       "      <td>8</td>\n",
       "      <td>100.00</td>\n",
       "      <td>129.75</td>\n",
       "      <td>6.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2G5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>340</td>\n",
       "      <td>129</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>29.07</td>\n",
       "      <td>2.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2G4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>457</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>3.68</td>\n",
       "      <td>13.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2F5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>443</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>6.58</td>\n",
       "      <td>12.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2C2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>424</td>\n",
       "      <td>45</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>9.25</td>\n",
       "      <td>1.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2B6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>369</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>95.91</td>\n",
       "      <td>5.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1G1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>0</td>\n",
       "      <td>445</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>6.84</td>\n",
       "      <td>6.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3I5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>464</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.91</td>\n",
       "      <td>1.90</td>\n",
       "      <td>14.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2B3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>463</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.86</td>\n",
       "      <td>2.04</td>\n",
       "      <td>40.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1I7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>423</td>\n",
       "      <td>43</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.73</td>\n",
       "      <td>10.80</td>\n",
       "      <td>6.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1H2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>393</td>\n",
       "      <td>65</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>99.73</td>\n",
       "      <td>15.82</td>\n",
       "      <td>7.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1F9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>452</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.73</td>\n",
       "      <td>4.04</td>\n",
       "      <td>10.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1E4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>134</td>\n",
       "      <td>150</td>\n",
       "      <td>88</td>\n",
       "      <td>38</td>\n",
       "      <td>99.73</td>\n",
       "      <td>182.60</td>\n",
       "      <td>1.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1A7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>463</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.64</td>\n",
       "      <td>1.59</td>\n",
       "      <td>14.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1H5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>421</td>\n",
       "      <td>48</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.59</td>\n",
       "      <td>9.37</td>\n",
       "      <td>7.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4B7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>104</td>\n",
       "      <td>111</td>\n",
       "      <td>111</td>\n",
       "      <td>75</td>\n",
       "      <td>69</td>\n",
       "      <td>99.46</td>\n",
       "      <td>163.90</td>\n",
       "      <td>1.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4A8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>3</td>\n",
       "      <td>110</td>\n",
       "      <td>108</td>\n",
       "      <td>116</td>\n",
       "      <td>65</td>\n",
       "      <td>70</td>\n",
       "      <td>99.46</td>\n",
       "      <td>161.98</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2G8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>434</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.46</td>\n",
       "      <td>8.62</td>\n",
       "      <td>3.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2C9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>394</td>\n",
       "      <td>71</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.46</td>\n",
       "      <td>15.30</td>\n",
       "      <td>6.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1I4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>168</td>\n",
       "      <td>99</td>\n",
       "      <td>110</td>\n",
       "      <td>63</td>\n",
       "      <td>30</td>\n",
       "      <td>99.46</td>\n",
       "      <td>126.31</td>\n",
       "      <td>6.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1H4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>460</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.46</td>\n",
       "      <td>2.54</td>\n",
       "      <td>10.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1D4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>2</td>\n",
       "      <td>153</td>\n",
       "      <td>245</td>\n",
       "      <td>55</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>99.46</td>\n",
       "      <td>84.11</td>\n",
       "      <td>1.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2E9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>3</td>\n",
       "      <td>441</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.37</td>\n",
       "      <td>5.41</td>\n",
       "      <td>9.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>1B6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>3</td>\n",
       "      <td>415</td>\n",
       "      <td>49</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.37</td>\n",
       "      <td>11.29</td>\n",
       "      <td>4.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>4C7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>3</td>\n",
       "      <td>365</td>\n",
       "      <td>94</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>99.18</td>\n",
       "      <td>23.38</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>4C1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>3</td>\n",
       "      <td>123</td>\n",
       "      <td>169</td>\n",
       "      <td>141</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>99.18</td>\n",
       "      <td>109.82</td>\n",
       "      <td>1.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>3A1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>11</td>\n",
       "      <td>337</td>\n",
       "      <td>102</td>\n",
       "      <td>17</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>97.28</td>\n",
       "      <td>35.19</td>\n",
       "      <td>82.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>1D8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>14</td>\n",
       "      <td>343</td>\n",
       "      <td>100</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>97.09</td>\n",
       "      <td>27.86</td>\n",
       "      <td>14.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>1B1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>12</td>\n",
       "      <td>392</td>\n",
       "      <td>57</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>97.06</td>\n",
       "      <td>17.07</td>\n",
       "      <td>62.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>2A8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>15</td>\n",
       "      <td>134</td>\n",
       "      <td>272</td>\n",
       "      <td>46</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>97.03</td>\n",
       "      <td>79.14</td>\n",
       "      <td>8.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>1C4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>15</td>\n",
       "      <td>358</td>\n",
       "      <td>84</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>96.97</td>\n",
       "      <td>21.49</td>\n",
       "      <td>14.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>3F1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>15</td>\n",
       "      <td>433</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>96.88</td>\n",
       "      <td>4.92</td>\n",
       "      <td>65.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>1C9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>14</td>\n",
       "      <td>350</td>\n",
       "      <td>89</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>96.88</td>\n",
       "      <td>29.35</td>\n",
       "      <td>75.31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>2B9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>15</td>\n",
       "      <td>373</td>\n",
       "      <td>79</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>96.78</td>\n",
       "      <td>16.91</td>\n",
       "      <td>12.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>1C1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>14</td>\n",
       "      <td>426</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>96.60</td>\n",
       "      <td>8.04</td>\n",
       "      <td>60.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>3F3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>15</td>\n",
       "      <td>407</td>\n",
       "      <td>46</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>96.56</td>\n",
       "      <td>9.78</td>\n",
       "      <td>22.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>1A2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>19</td>\n",
       "      <td>348</td>\n",
       "      <td>85</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>95.97</td>\n",
       "      <td>25.82</td>\n",
       "      <td>43.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>3B3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>19</td>\n",
       "      <td>371</td>\n",
       "      <td>73</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95.88</td>\n",
       "      <td>19.29</td>\n",
       "      <td>14.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>3G5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>18</td>\n",
       "      <td>411</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95.65</td>\n",
       "      <td>8.17</td>\n",
       "      <td>47.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>2A9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>19</td>\n",
       "      <td>371</td>\n",
       "      <td>75</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>95.56</td>\n",
       "      <td>16.02</td>\n",
       "      <td>12.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>2C4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>19</td>\n",
       "      <td>429</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95.43</td>\n",
       "      <td>4.35</td>\n",
       "      <td>66.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>3E6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>20</td>\n",
       "      <td>406</td>\n",
       "      <td>45</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95.15</td>\n",
       "      <td>9.60</td>\n",
       "      <td>37.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>3C2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>21</td>\n",
       "      <td>219</td>\n",
       "      <td>152</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>95.11</td>\n",
       "      <td>69.41</td>\n",
       "      <td>11.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>3I7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>25</td>\n",
       "      <td>323</td>\n",
       "      <td>102</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>94.93</td>\n",
       "      <td>33.51</td>\n",
       "      <td>89.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>2D2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>26</td>\n",
       "      <td>200</td>\n",
       "      <td>152</td>\n",
       "      <td>73</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>94.64</td>\n",
       "      <td>77.64</td>\n",
       "      <td>37.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>4C4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>24</td>\n",
       "      <td>128</td>\n",
       "      <td>103</td>\n",
       "      <td>112</td>\n",
       "      <td>62</td>\n",
       "      <td>43</td>\n",
       "      <td>94.38</td>\n",
       "      <td>137.91</td>\n",
       "      <td>1.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>3E2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>39</td>\n",
       "      <td>402</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>92.53</td>\n",
       "      <td>7.49</td>\n",
       "      <td>60.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>4A3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>33</td>\n",
       "      <td>198</td>\n",
       "      <td>184</td>\n",
       "      <td>50</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>92.39</td>\n",
       "      <td>66.57</td>\n",
       "      <td>83.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>2D1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>44</td>\n",
       "      <td>145</td>\n",
       "      <td>142</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>15</td>\n",
       "      <td>91.97</td>\n",
       "      <td>97.71</td>\n",
       "      <td>36.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>2I5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>33</td>\n",
       "      <td>360</td>\n",
       "      <td>57</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>91.94</td>\n",
       "      <td>23.39</td>\n",
       "      <td>86.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>2H7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>42</td>\n",
       "      <td>138</td>\n",
       "      <td>198</td>\n",
       "      <td>81</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>91.55</td>\n",
       "      <td>86.72</td>\n",
       "      <td>8.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>3C4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>41</td>\n",
       "      <td>228</td>\n",
       "      <td>145</td>\n",
       "      <td>39</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>91.55</td>\n",
       "      <td>54.66</td>\n",
       "      <td>11.73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>4D6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>43</td>\n",
       "      <td>345</td>\n",
       "      <td>75</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>90.76</td>\n",
       "      <td>16.19</td>\n",
       "      <td>22.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>4B2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>44</td>\n",
       "      <td>408</td>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>90.51</td>\n",
       "      <td>3.80</td>\n",
       "      <td>31.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>4G2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>41</td>\n",
       "      <td>301</td>\n",
       "      <td>110</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>90.26</td>\n",
       "      <td>32.60</td>\n",
       "      <td>85.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>4C8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>40</td>\n",
       "      <td>102</td>\n",
       "      <td>134</td>\n",
       "      <td>132</td>\n",
       "      <td>51</td>\n",
       "      <td>13</td>\n",
       "      <td>90.04</td>\n",
       "      <td>121.17</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   bin_id marker_lineage marker_number  genomes  markers  marker_sets   0  \\\n",
       "0     3C6  Cyanobacteria           (1)      129      472          368   0   \n",
       "1     2I9  Cyanobacteria           (1)      129      472          368   0   \n",
       "2     2I6  Cyanobacteria           (1)      129      472          368   0   \n",
       "3     2H1  Cyanobacteria           (1)      129      472          368   0   \n",
       "4     2G7  Cyanobacteria           (1)      129      472          368   0   \n",
       "5     2G5  Cyanobacteria           (1)      129      472          368   0   \n",
       "6     2G4  Cyanobacteria           (1)      129      472          368   0   \n",
       "7     2F5  Cyanobacteria           (1)      129      472          368   0   \n",
       "8     2C2  Cyanobacteria           (1)      129      472          368   0   \n",
       "9     2B6  Cyanobacteria           (1)      129      472          368   0   \n",
       "10    1G1  Cyanobacteria           (1)      129      472          368   0   \n",
       "11    3I5  Cyanobacteria           (1)      129      472          368   1   \n",
       "12    2B3  Cyanobacteria           (1)      129      472          368   1   \n",
       "13    1I7  Cyanobacteria           (1)      129      472          368   1   \n",
       "14    1H2  Cyanobacteria           (1)      129      472          368   1   \n",
       "15    1F9  Cyanobacteria           (1)      129      472          368   1   \n",
       "16    1E4  Cyanobacteria           (1)      129      472          368   1   \n",
       "17    1A7  Cyanobacteria           (1)      129      472          368   2   \n",
       "18    1H5  Cyanobacteria           (1)      129      472          368   2   \n",
       "19    4B7  Cyanobacteria           (1)      129      472          368   2   \n",
       "20    4A8  Cyanobacteria           (1)      129      472          368   3   \n",
       "21    2G8  Cyanobacteria           (1)      129      472          368   2   \n",
       "22    2C9  Cyanobacteria           (1)      129      472          368   2   \n",
       "23    1I4  Cyanobacteria           (1)      129      472          368   2   \n",
       "24    1H4  Cyanobacteria           (1)      129      472          368   2   \n",
       "25    1D4  Cyanobacteria           (1)      129      472          368   2   \n",
       "26    2E9  Cyanobacteria           (1)      129      472          368   3   \n",
       "27    1B6  Cyanobacteria           (1)      129      472          368   3   \n",
       "28    4C7  Cyanobacteria           (1)      129      472          368   3   \n",
       "29    4C1  Cyanobacteria           (1)      129      472          368   3   \n",
       "..    ...            ...           ...      ...      ...          ...  ..   \n",
       "67    3A1  Cyanobacteria           (1)      129      472          368  11   \n",
       "68    1D8  Cyanobacteria           (1)      129      472          368  14   \n",
       "69    1B1  Cyanobacteria           (1)      129      472          368  12   \n",
       "70    2A8  Cyanobacteria           (1)      129      472          368  15   \n",
       "71    1C4  Cyanobacteria           (1)      129      472          368  15   \n",
       "72    3F1  Cyanobacteria           (1)      129      472          368  15   \n",
       "73    1C9  Cyanobacteria           (1)      129      472          368  14   \n",
       "74    2B9  Cyanobacteria           (1)      129      472          368  15   \n",
       "75    1C1  Cyanobacteria           (1)      129      472          368  14   \n",
       "76    3F3  Cyanobacteria           (1)      129      472          368  15   \n",
       "77    1A2  Cyanobacteria           (1)      129      472          368  19   \n",
       "78    3B3  Cyanobacteria           (1)      129      472          368  19   \n",
       "79    3G5  Cyanobacteria           (1)      129      472          368  18   \n",
       "80    2A9  Cyanobacteria           (1)      129      472          368  19   \n",
       "81    2C4  Cyanobacteria           (1)      129      472          368  19   \n",
       "82    3E6  Cyanobacteria           (1)      129      472          368  20   \n",
       "83    3C2  Cyanobacteria           (1)      129      472          368  21   \n",
       "84    3I7  Cyanobacteria           (1)      129      472          368  25   \n",
       "85    2D2  Cyanobacteria           (1)      129      472          368  26   \n",
       "86    4C4  Cyanobacteria           (1)      129      472          368  24   \n",
       "87    3E2  Cyanobacteria           (1)      129      472          368  39   \n",
       "88    4A3  Cyanobacteria           (1)      129      472          368  33   \n",
       "89    2D1  Cyanobacteria           (1)      129      472          368  44   \n",
       "90    2I5  Cyanobacteria           (1)      129      472          368  33   \n",
       "91    2H7  Cyanobacteria           (1)      129      472          368  42   \n",
       "92    3C4  Cyanobacteria           (1)      129      472          368  41   \n",
       "93    4D6  Cyanobacteria           (1)      129      472          368  43   \n",
       "94    4B2  Cyanobacteria           (1)      129      472          368  44   \n",
       "95    4G2  Cyanobacteria           (1)      129      472          368  41   \n",
       "96    4C8  Cyanobacteria           (1)      129      472          368  40   \n",
       "\n",
       "      1    2    3   4  5+  completeness  contamination  strain_heterogeneity  \n",
       "0   380   89    3   0   0        100.00          18.57                  8.16  \n",
       "1     0  411   52   6   3        100.00         114.24                 46.24  \n",
       "2     4  220  179  57  12        100.00         166.73                  4.86  \n",
       "3     3  117  299  47   6        100.00         185.79                  4.72  \n",
       "4    58  259  112  35   8        100.00         129.75                  6.37  \n",
       "5   340  129    3   0   0        100.00          29.07                  2.90  \n",
       "6   457   15    0   0   0        100.00           3.68                 13.33  \n",
       "7   443   27    2   0   0        100.00           6.58                 12.12  \n",
       "8   424   45    3   0   0        100.00           9.25                  1.85  \n",
       "9    62  369   40   1   0        100.00          95.91                  5.25  \n",
       "10  445   26    1   0   0        100.00           6.84                  6.90  \n",
       "11  464    7    0   0   0         99.91           1.90                 14.29  \n",
       "12  463    7    1   0   0         99.86           2.04                 40.00  \n",
       "13  423   43    4   1   0         99.73          10.80                  6.56  \n",
       "14  393   65   10   3   0         99.73          15.82                  7.96  \n",
       "15  452   19    0   0   0         99.73           4.04                 10.53  \n",
       "16   61  134  150  88  38         99.73         182.60                  1.03  \n",
       "17  463    7    0   0   0         99.64           1.59                 14.29  \n",
       "18  421   48    1   0   0         99.59           9.37                  7.84  \n",
       "19  104  111  111  75  69         99.46         163.90                  1.27  \n",
       "20  110  108  116  65  70         99.46         161.98                  0.98  \n",
       "21  434   28    8   0   0         99.46           8.62                  3.85  \n",
       "22  394   71    4   1   0         99.46          15.30                  6.74  \n",
       "23  168   99  110  63  30         99.46         126.31                  6.17  \n",
       "24  460   10    0   0   0         99.46           2.54                 10.00  \n",
       "25  153  245   55  14   3         99.46          84.11                  1.53  \n",
       "26  441   26    2   0   0         99.37           5.41                  9.38  \n",
       "27  415   49    5   0   0         99.37          11.29                  4.69  \n",
       "28  365   94    8   2   0         99.18          23.38                  0.00  \n",
       "29  123  169  141  22  14         99.18         109.82                  1.36  \n",
       "..  ...  ...  ...  ..  ..           ...            ...                   ...  \n",
       "67  337  102   17   5   0         97.28          35.19                 82.51  \n",
       "68  343  100   11   4   0         97.09          27.86                 14.01  \n",
       "69  392   57   10   1   0         97.06          17.07                 62.37  \n",
       "70  134  272   46   3   2         97.03          79.14                  8.30  \n",
       "71  358   84   12   3   0         96.97          21.49                 14.49  \n",
       "72  433   23    1   0   0         96.88           4.92                 65.38  \n",
       "73  350   89   15   3   1         96.88          29.35                 75.31  \n",
       "74  373   79    5   0   0         96.78          16.91                 12.77  \n",
       "75  426   28    4   0   0         96.60           8.04                 60.00  \n",
       "76  407   46    3   1   0         96.56           9.78                 22.95  \n",
       "77  348   85   18   2   0         95.97          25.82                 43.05  \n",
       "78  371   73    9   0   0         95.88          19.29                 14.00  \n",
       "79  411   39    4   0   0         95.65           8.17                 47.06  \n",
       "80  371   75    6   1   0         95.56          16.02                 12.12  \n",
       "81  429   24    0   0   0         95.43           4.35                 66.67  \n",
       "82  406   45    1   0   0         95.15           9.60                 37.50  \n",
       "83  219  152   64  12   4         95.11          69.41                 11.18  \n",
       "84  323  102   22   0   0         94.93          33.51                 89.88  \n",
       "85  200  152   73  18   3         94.64          77.64                 37.52  \n",
       "86  128  103  112  62  43         94.38         137.91                  1.59  \n",
       "87  402   30    1   0   0         92.53           7.49                 60.61  \n",
       "88  198  184   50   7   0         92.39          66.57                 83.78  \n",
       "89  145  142   92  34  15         91.97          97.71                 36.32  \n",
       "90  360   57   15   7   0         91.94          23.39                 86.11  \n",
       "91  138  198   81  11   2         91.55          86.72                  8.92  \n",
       "92  228  145   39  15   4         91.55          54.66                 11.73  \n",
       "93  345   75    8   1   0         90.76          16.19                 22.86  \n",
       "94  408   19    1   0   0         90.51           3.80                 31.82  \n",
       "95  301  110   20   0   0         90.26          32.60                 85.88  \n",
       "96  102  134  132  51  13         90.04         121.17                  0.98  \n",
       "\n",
       "[97 rows x 15 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkm_df1[checkm_df1.completeness >= 90]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Adding genomic IDs**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3LMooreaBCUL</td>\n",
       "      <td>4B7</td>\n",
       "      <td>99.46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3LOscillatoriaCUL</td>\n",
       "      <td>4B8</td>\n",
       "      <td>99.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASF14JUL141CUL(1a-algae)</td>\n",
       "      <td>4B3</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASG15JUL146CUL</td>\n",
       "      <td>3F2</td>\n",
       "      <td>99.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASI16JUL142CUL</td>\n",
       "      <td>1E3</td>\n",
       "      <td>98.91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               collectionID genomeID  completeness\n",
       "0              3LMooreaBCUL      4B7         99.46\n",
       "1         3LOscillatoriaCUL      4B8         99.18\n",
       "2  ASF14JUL141CUL(1a-algae)      4B3          0.00\n",
       "3            ASG15JUL146CUL      3F2         99.09\n",
       "4            ASI16JUL142CUL      1E3         98.91"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df = pd.read_csv(genome_loci, sep='\\t')\n",
    "count = 0\n",
    "checkm_col = []\n",
    "\n",
    "\n",
    "for i,r in summary_df.iterrows():\n",
    "    if len(summary_df.genomeID.loc[i]) < 3:\n",
    "        summary_df.genomeID.loc[i] = \"1%s\"%summary_df.genomeID.loc[i]\n",
    "    if summary_df.genomeID.loc[i] in checkm_dict.keys():\n",
    "        count += 1\n",
    "        checkm_col.append(checkm_dict[summary_df.genomeID.loc[i]])\n",
    "    else:\n",
    "        checkm_col.append(\"NA\")\n",
    "\n",
    "summary_df[\"completeness\"] = checkm_col\n",
    "\n",
    "summary_df = summary_df.replace(\"NA\",0) #zero completeness means the genome did not get scaffold\n",
    "\n",
    "summary_df[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(161, 265)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(checkm_df1),len(summary_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(96, 3)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df[summary_df.completeness >= 90].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Renaming and copy scaffolded genomes with high quality**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir /home/tiago/Desktop/cyanet/cyanobiome/scaffolds/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def rename_scaffolds(fasta_file,outpath):\n",
    "    strain = os.path.basename(fasta_file).split(\".\")[0]\n",
    "    new_file = []\n",
    "    input_handle = open(fasta_file, \"rU\")\n",
    "    node_count = 1\n",
    "    for record in SeqIO.parse(input_handle, \"fasta\"):\n",
    "        new_id = \"%s_NODE_%s\"%(strain,node_count)\n",
    "        record.id = str(new_id)\n",
    "        record.description = \"\"\n",
    "        new_file.append(record)\n",
    "        node_count += 1\n",
    "    output_handle = open(\"%s/%s.fasta\"%(outpath,strain), \"w\")\n",
    "    SeqIO.write(new_file, output_handle, \"fasta\")\n",
    "    output_handle.close()\n",
    "    input_handle.close()\n",
    "    print \"%s contigs renamed\"%(node_count-1)\n",
    "\n",
    "scf_list = list(summary_df[summary_df.completeness >= 90][\"genomeID\"])    \n",
    "##removing previously published and re-sequenced genomes\n",
    "scf_list.remove(\"4C7\") \n",
    "scf_list.remove(\"4B8\")\n",
    "scf_list.remove(\"4C8\")\n",
    "scf_list.remove(\"3F3\")\n",
    "scf_list.remove(\"4B7\")\n",
    "scf_list.remove(\"1E3\")\n",
    "\n",
    "##removing highly fragmened genomes\n",
    "scf_list.remove(\"2D8\")\n",
    "scf_list.remove(\"1B8\")\n",
    "scf_list.remove(\"1C1\")\n",
    "scf_list.remove(\"3F1\")\n",
    "scf_list.remove(\"1C9\")\n",
    "scf_list.remove(\"4B2\")\n",
    "scf_list.remove(\"4A8\")\n",
    "\n",
    "##removing genomes with more than one cyanobacteria\n",
    "scf_list.remove(\"2I9\")\n",
    "scf_list.remove(\"2H1\")\n",
    "scf_list.remove(\"2I6\")\n",
    "scf_list.remove(\"1D4\")\n",
    "scf_list.remove(\"1I4\")\n",
    "\n",
    "##removing duplicated genomes (cyanobacteria was cultured)\n",
    "scf_list.remove(\"1B6\")\n",
    "scf_list.remove(\"1A2\")\n",
    "scf_list.remove(\"2A9\")\n",
    "scf_list.remove(\"2G8\")\n",
    "scf_list.remove(\"3A1\")\n",
    "\n",
    "\n",
    "for item in scf_list:\n",
    "    scaffold = \"/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/%s.fasta\"%item\n",
    "    rename_scaffolds(scaffold,\"/home/tiago/Desktop/cyanet/cyanobiome/scaffolds/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Removing duplicates from missing list**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "105"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_df = summary_df[summary_df.completeness == 0]\n",
    "\n",
    "len(missing_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4D7.fasta\n",
      "/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4B1.fasta\n",
      "/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A6.fasta\n",
      "/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E4.fasta\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "101"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_out = []\n",
    "\n",
    "for i,r in missing_df.iterrows():\n",
    "    genomeID = missing_df.genomeID.loc[i]\n",
    "    scaffolded_genome = glob.glob(\"/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/%s*fasta\"%genomeID)\n",
    "    if scaffolded_genome:\n",
    "        print scaffolded_genome[0]\n",
    "        index_out.append(i)\n",
    "        \n",
    "missing_df = missing_df.drop(index_out)\n",
    "\n",
    "len(missing_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**PS: these 4 genomes will be included in the next CheckM run**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Making darkhorse list**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "smry_list = glob.glob(\"/home/gerwick-lab/Desktop/data/genomes/prokka/darkhorse/*/*_smry\")\n",
    "smry_dict = {}\n",
    "\n",
    "for item in smry_list:\n",
    "    logfile = os.path.join(os.path.dirname(item),os.path.basename(item).split(\"_\")[0]+\"_logfile\")\n",
    "    f=open(logfile)\n",
    "    lines=f.readlines()\n",
    "    m = re.search(r'(\\S*).matches.m8',lines[4])\n",
    "    if m:\n",
    "        smry_dict[m.group(1)] = item\n",
    "        \n",
    "len(smry_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Binning metagenomic contigs**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def loading_darkhorse_taxa_dict(smry_file):\n",
    "    darkhorse_df = pd.read_csv(smry_file, sep='\\t')\n",
    "    taxa_dict = {}\n",
    "    for i,r in darkhorse_df.iterrows():\n",
    "        taxa_class = str(darkhorse_df.lineage.loc[i].split(\";\",2)[1]).split(\" \",1)[0]\n",
    "        taxa_dict[darkhorse_df.query_id.loc[i]] = taxa_class\n",
    "    return taxa_dict\n",
    "\n",
    "def loading_prokka_to_contig_mapping(filename):\n",
    "    prokka_id_to_contig_mapping = {}\n",
    "    contig_id = \"\"\n",
    "    for line in open(filename):\n",
    "        if line.find(\">Feature\") != -1:\n",
    "            contig_id = line.rstrip().split(\" \")[1]\n",
    "            continue\n",
    "        if line.find(\"locus_tag\") != -1:\n",
    "            prokka_id = line.lstrip().rstrip().split(\"\\t\")[-1]\n",
    "            prokka_id_to_contig_mapping[prokka_id] = contig_id\n",
    "    return prokka_id_to_contig_mapping\n",
    "            \n",
    "def get_contig_list(smry_file,tbl_file):\n",
    "    contig_dict = loading_prokka_to_contig_mapping(tbl_file)\n",
    "    taxa_dict = loading_darkhorse_taxa_dict(smry_file)\n",
    "    merged_dict = {}\n",
    "    for key in contig_dict.iterkeys():\n",
    "        if contig_dict[key] not in merged_dict:\n",
    "            if key in taxa_dict:\n",
    "                merged_dict[contig_dict[key]] = taxa_dict[key]\n",
    "        else:\n",
    "            if key in taxa_dict:\n",
    "                merged_dict[contig_dict[key]] = merged_dict[contig_dict[key]] +\", \"+ taxa_dict[key]\n",
    "    contig_list = []\n",
    "    for key,value in merged_dict.items():\n",
    "        if \"Cyanobacteria\" in value:\n",
    "            contig_list.append(key)\n",
    "    return contig_list\n",
    "\n",
    "def save_binned_fasta(smry_file,tbl_file,input_fasta,output_fasta):\n",
    "    contig_list = get_contig_list(smry_file,tbl_file)\n",
    "    new_file = []\n",
    "    input_handle = open(input_fasta, \"rU\")\n",
    "    count = 1\n",
    "    for record in SeqIO.parse(input_handle, \"fasta\"):\n",
    "        if record.id in contig_list:\n",
    "            new_file.append(record)\n",
    "            count += 1\n",
    "    output_handle = open(output_fasta, \"w\")\n",
    "    SeqIO.write(new_file, output_handle, \"fasta\")\n",
    "    output_handle.close()\n",
    "    input_handle.close()\n",
    "    print \"%s records processed and saved into %s\"%(count,output_fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "###copying the hybrid assembly (ASI-nathan notebook) for ASI\n",
    "# !cp /home/tiago/Desktop/cyanet/asi-nathan/ASIH.ren.fasta /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/metagenomes/renamed/ASIH.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/metagenomes/renamed/ASIH.fasta\n"
     ]
    }
   ],
   "source": [
    "!ls /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/metagenomes/renamed/ASIH.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>ASI16JUL142CUL</td>\n",
       "      <td>ASIH</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       collectionID genomeID completeness\n",
       "101  ASI16JUL142CUL     ASIH            0"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#adding ASIH to missing_df\n",
    "\n",
    "df_ASIH = pd.DataFrame(columns=['collectionID', 'genomeID', 'completeness'])\n",
    "\n",
    "df_ASIH.loc[0] = [\"ASI16JUL142CUL\",\"ASIH\",0]\n",
    "\n",
    "missing_df = missing_df.append(df_ASIH,ignore_index=True)\n",
    "\n",
    "missing_df[missing_df.genomeID == \"ASIH\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "output_folder = \"/home/tiago/Desktop/cyanet/cyanobiome/meta_binned/\"\n",
    "no_sequence = []\n",
    "\n",
    "for i,r in missing_df.iterrows():\n",
    "    genomeID = missing_df.genomeID.loc[i]\n",
    "    metagen_file = glob.glob(\"/home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/metagenomes/renamed/%s*fasta\"%genomeID)\n",
    "    if metagen_file:\n",
    "        if genomeID in smry_dict:\n",
    "            tbl_file = glob.glob(\"/home/gerwick-lab/Desktop/data/genomes/prokka/annotation/%s/%s.tbl\"%(genomeID,genomeID))\n",
    "            if len(tbl_file) == 0:\n",
    "                tbl_file = glob.glob(\"/home/gerwick-lab/Desktop/data/genomes/prokka/annotation/%s/%s.tbl\"%(genomeID[1:3],genomeID[1:3]))\n",
    "            smry_file = smry_dict[genomeID]\n",
    "            output_fasta = output_folder + \"%s.fasta\"%genomeID\n",
    "            save_binned_fasta(smry_file,tbl_file[0],metagen_file[0],output_fasta)\n",
    "    else:\n",
    "        no_sequence.append(genomeID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(no_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!ls /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/*fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Adding missing genomes to folder**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4D7.fasta /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/\n",
    "!cp /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4B1.fasta /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/\n",
    "!cp /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4A6.fasta /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/\n",
    "!cp /home/gerwick-lab/Desktop/data/genomes/drafts/kappahyperplus/scaffolded_drafts/4E4.fasta /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/\n",
    "!cp /home/gerwick-lab/Desktop/data/genomes/complete/ISB-B.fasta /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!ls /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/*fasta | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Running second CheckM on binned metagenomes**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir /home/tiago/Desktop/cyanet/cyanobiome/tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "checkm taxonomy_wf phylum Cyanobacteria -x fasta \\\n",
    "/home/tiago/Desktop/cyanet/cyanobiome/meta_binned/ \\\n",
    "/home/tiago/Desktop/cyanet/cyanobiome/tables/ >> \\\n",
    "/home/tiago/Desktop/cyanet/cyanobiome/tables/checkM_metabinned_TFL190315\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def get_checkm_data(filename):\n",
    "    headers=['bin_id','marker_lineage','marker_number','genomes','markers','marker_sets','0','1','2','3','4','5+','completeness','contamination','strain_heterogeneity']\n",
    "    df = pd.read_csv(filename, delimiter=r'\\s+',names=headers,skiprows=3)\n",
    "    df = df.drop(df.index[len(df)-1])\n",
    "    return df\n",
    "\n",
    "checkm_df2 = get_checkm_data(\"/home/tiago/Desktop/cyanet/cyanobiome/tables/checkM_metabinned_TFL190315\")\n",
    "checkm_dict2 = pd.Series(checkm_df2[\"completeness\"].values,index=checkm_df2[\"bin_id\"]).to_dict()\n",
    "\n",
    "merged_checkm = checkm_dict.copy()\n",
    "merged_checkm.update(checkm_dict2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(161, 37, 198)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(checkm_dict),len(checkm_dict2),len(merged_checkm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_id</th>\n",
       "      <th>marker_lineage</th>\n",
       "      <th>marker_number</th>\n",
       "      <th>genomes</th>\n",
       "      <th>markers</th>\n",
       "      <th>marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4A6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>418.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>110.43</td>\n",
       "      <td>3.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASIH</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>432.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>98.91</td>\n",
       "      <td>6.68</td>\n",
       "      <td>4.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISB-B</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>460.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>98.82</td>\n",
       "      <td>1.77</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4B3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>76.45</td>\n",
       "      <td>96.45</td>\n",
       "      <td>0.89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3F8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>148.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>71.60</td>\n",
       "      <td>75.91</td>\n",
       "      <td>2.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2B5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>58.15</td>\n",
       "      <td>5.41</td>\n",
       "      <td>25.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2E1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>281.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.15</td>\n",
       "      <td>2.92</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3A9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>337.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.81</td>\n",
       "      <td>5.48</td>\n",
       "      <td>5.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4E6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>412.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.83</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3C9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>407.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3H9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>424.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.44</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3F6</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>418.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.84</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3G8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>433.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.53</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3D1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>444.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.73</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3H4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>442.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.62</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4F2</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>449.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.63</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2B1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>451.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.21</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2D7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>447.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.88</td>\n",
       "      <td>0.68</td>\n",
       "      <td>33.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>3E4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.48</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4F5</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>456.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.82</td>\n",
       "      <td>0.01</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2G9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.93</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2D9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>461.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.70</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4F9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>464.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.64</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3B9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>466.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.36</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4E9</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>463.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.29</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>4E4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3A3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>465.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3D3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>468.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>3I3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>470.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>3C7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>468.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>4F1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>464.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>3I1</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>470.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>3B4</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>471.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>4F8</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>3E3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>3C3</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>2C7</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>(1)</td>\n",
       "      <td>129.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   bin_id marker_lineage marker_number  genomes  markers  marker_sets      0  \\\n",
       "0     4A6  Cyanobacteria           (1)    129.0    472.0        368.0    0.0   \n",
       "1    ASIH  Cyanobacteria           (1)    129.0    472.0        368.0    4.0   \n",
       "2   ISB-B  Cyanobacteria           (1)    129.0    472.0        368.0    5.0   \n",
       "3     4B3  Cyanobacteria           (1)    129.0    472.0        368.0   91.0   \n",
       "4     3F8  Cyanobacteria           (1)    129.0    472.0        368.0  111.0   \n",
       "5     2B5  Cyanobacteria           (1)    129.0    472.0        368.0  169.0   \n",
       "6     2E1  Cyanobacteria           (1)    129.0    472.0        368.0  281.0   \n",
       "7     3A9  Cyanobacteria           (1)    129.0    472.0        368.0  337.0   \n",
       "8     4E6  Cyanobacteria           (1)    129.0    472.0        368.0  412.0   \n",
       "9     3C9  Cyanobacteria           (1)    129.0    472.0        368.0  407.0   \n",
       "10    3H9  Cyanobacteria           (1)    129.0    472.0        368.0  424.0   \n",
       "11    3F6  Cyanobacteria           (1)    129.0    472.0        368.0  418.0   \n",
       "12    3G8  Cyanobacteria           (1)    129.0    472.0        368.0  433.0   \n",
       "13    3D1  Cyanobacteria           (1)    129.0    472.0        368.0  444.0   \n",
       "14    3H4  Cyanobacteria           (1)    129.0    472.0        368.0  442.0   \n",
       "15    4F2  Cyanobacteria           (1)    129.0    472.0        368.0  449.0   \n",
       "16    2B1  Cyanobacteria           (1)    129.0    472.0        368.0  451.0   \n",
       "17    2D7  Cyanobacteria           (1)    129.0    472.0        368.0  447.0   \n",
       "18    3E4  Cyanobacteria           (1)    129.0    472.0        368.0  450.0   \n",
       "19    4F5  Cyanobacteria           (1)    129.0    472.0        368.0  456.0   \n",
       "20    2G9  Cyanobacteria           (1)    129.0    472.0        368.0  462.0   \n",
       "21    2D9  Cyanobacteria           (1)    129.0    472.0        368.0  461.0   \n",
       "22    4F9  Cyanobacteria           (1)    129.0    472.0        368.0  464.0   \n",
       "23    3B9  Cyanobacteria           (1)    129.0    472.0        368.0  466.0   \n",
       "24    4E9  Cyanobacteria           (1)    129.0    472.0        368.0  463.0   \n",
       "25    4E4  Cyanobacteria           (1)    129.0    472.0        368.0  462.0   \n",
       "26    3A3  Cyanobacteria           (1)    129.0    472.0        368.0  465.0   \n",
       "27    3D3  Cyanobacteria           (1)    129.0    472.0        368.0  468.0   \n",
       "28    3I3  Cyanobacteria           (1)    129.0    472.0        368.0  470.0   \n",
       "29    3C7  Cyanobacteria           (1)    129.0    472.0        368.0  468.0   \n",
       "30    4F1  Cyanobacteria           (1)    129.0    472.0        368.0  464.0   \n",
       "31    3I1  Cyanobacteria           (1)    129.0    472.0        368.0  470.0   \n",
       "32    3B4  Cyanobacteria           (1)    129.0    472.0        368.0  471.0   \n",
       "33    4F8  Cyanobacteria           (1)    129.0    472.0        368.0  472.0   \n",
       "34    3E3  Cyanobacteria           (1)    129.0    472.0        368.0  472.0   \n",
       "35    3C3  Cyanobacteria           (1)    129.0    472.0        368.0  472.0   \n",
       "36    2C7  Cyanobacteria           (1)    129.0    472.0        368.0  472.0   \n",
       "\n",
       "        1      2     3     4    5+  completeness  contamination  \\\n",
       "0     3.0  418.0  45.0   6.0   0.0        100.00         110.43   \n",
       "1   432.0   30.0   6.0   0.0   0.0         98.91           6.68   \n",
       "2   460.0    6.0   0.0   1.0   0.0         98.82           1.77   \n",
       "3    74.0  165.0  87.0  39.0  16.0         76.45          96.45   \n",
       "4   148.0  128.0  43.0  29.0  13.0         71.60          75.91   \n",
       "5   275.0   26.0   1.0   1.0   0.0         58.15           5.41   \n",
       "6   161.0   30.0   0.0   0.0   0.0         35.15           2.92   \n",
       "7   106.0   24.0   5.0   0.0   0.0         26.81           5.48   \n",
       "8    60.0    0.0   0.0   0.0   0.0         12.83           0.00   \n",
       "9    65.0    0.0   0.0   0.0   0.0         12.16           0.00   \n",
       "10   48.0    0.0   0.0   0.0   0.0         10.44           0.00   \n",
       "11   54.0    0.0   0.0   0.0   0.0          9.84           0.00   \n",
       "12   38.0    1.0   0.0   0.0   0.0          7.53           0.27   \n",
       "13   28.0    0.0   0.0   0.0   0.0          5.73           0.00   \n",
       "14   30.0    0.0   0.0   0.0   0.0          5.62           0.00   \n",
       "15   23.0    0.0   0.0   0.0   0.0          4.63           0.00   \n",
       "16   20.0    1.0   0.0   0.0   0.0          4.21           0.09   \n",
       "17   22.0    3.0   0.0   0.0   0.0          3.88           0.68   \n",
       "18   20.0    2.0   0.0   0.0   0.0          3.48           0.06   \n",
       "19   15.0    1.0   0.0   0.0   0.0          2.82           0.01   \n",
       "20   10.0    0.0   0.0   0.0   0.0          1.93           0.00   \n",
       "21   11.0    0.0   0.0   0.0   0.0          1.70           0.00   \n",
       "22    8.0    0.0   0.0   0.0   0.0          1.64           0.00   \n",
       "23    6.0    0.0   0.0   0.0   0.0          1.36           0.00   \n",
       "24    9.0    0.0   0.0   0.0   0.0          1.29           0.00   \n",
       "25   10.0    0.0   0.0   0.0   0.0          1.28           0.00   \n",
       "26    7.0    0.0   0.0   0.0   0.0          0.86           0.00   \n",
       "27    4.0    0.0   0.0   0.0   0.0          0.59           0.00   \n",
       "28    2.0    0.0   0.0   0.0   0.0          0.54           0.00   \n",
       "29    4.0    0.0   0.0   0.0   0.0          0.43           0.00   \n",
       "30    8.0    0.0   0.0   0.0   0.0          0.42           0.00   \n",
       "31    2.0    0.0   0.0   0.0   0.0          0.15           0.00   \n",
       "32    1.0    0.0   0.0   0.0   0.0          0.09           0.00   \n",
       "33    0.0    0.0   0.0   0.0   0.0          0.00           0.00   \n",
       "34    0.0    0.0   0.0   0.0   0.0          0.00           0.00   \n",
       "35    0.0    0.0   0.0   0.0   0.0          0.00           0.00   \n",
       "36    0.0    0.0   0.0   0.0   0.0          0.00           0.00   \n",
       "\n",
       "    strain_heterogeneity  \n",
       "0                   3.06  \n",
       "1                   4.17  \n",
       "2                   0.00  \n",
       "3                   0.89  \n",
       "4                   2.54  \n",
       "5                  25.71  \n",
       "6                   0.00  \n",
       "7                   5.13  \n",
       "8                   0.00  \n",
       "9                   0.00  \n",
       "10                  0.00  \n",
       "11                  0.00  \n",
       "12                  0.00  \n",
       "13                  0.00  \n",
       "14                  0.00  \n",
       "15                  0.00  \n",
       "16                  0.00  \n",
       "17                 33.33  \n",
       "18                  0.00  \n",
       "19                100.00  \n",
       "20                  0.00  \n",
       "21                  0.00  \n",
       "22                  0.00  \n",
       "23                  0.00  \n",
       "24                  0.00  \n",
       "25                  0.00  \n",
       "26                  0.00  \n",
       "27                  0.00  \n",
       "28                  0.00  \n",
       "29                  0.00  \n",
       "30                  0.00  \n",
       "31                  0.00  \n",
       "32                  0.00  \n",
       "33                  0.00  \n",
       "34                  0.00  \n",
       "35                  0.00  \n",
       "36                  0.00  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkm_df2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Copying high-quality genomes to scaffolds folder**\n",
    "```\n",
    "The only two high-quality genomes outside scaffolds folder was ISB-B (single contig manually renamed as ISBB_NODE_1) and ASIH, 4A6 is very complete but also very contaminated\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!mv /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/ISB-B.fasta /home/tiago/Desktop/cyanet/cyanobiome/scaffolds/ISBB.fasta\n",
    "!mv /home/tiago/Desktop/cyanet/cyanobiome/meta_binned/ASIH.fasta /home/tiago/Desktop/cyanet/cyanobiome/scaffolds/ASIH.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Generating master dataframe**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3LMooreaBCUL</td>\n",
       "      <td>4B7</td>\n",
       "      <td>99.46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3LOscillatoriaCUL</td>\n",
       "      <td>4B8</td>\n",
       "      <td>99.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASF14JUL141CUL(1a-algae)</td>\n",
       "      <td>4B3</td>\n",
       "      <td>76.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASG15JUL146CUL</td>\n",
       "      <td>3F2</td>\n",
       "      <td>99.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASI16JUL142CUL</td>\n",
       "      <td>1E3</td>\n",
       "      <td>98.91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               collectionID genomeID  completeness\n",
       "0              3LMooreaBCUL      4B7         99.46\n",
       "1         3LOscillatoriaCUL      4B8         99.18\n",
       "2  ASF14JUL141CUL(1a-algae)      4B3         76.45\n",
       "3            ASG15JUL146CUL      3F2         99.09\n",
       "4            ASI16JUL142CUL      1E3         98.91"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df = pd.read_csv(genome_loci, sep='\\t')\n",
    "count = 0\n",
    "checkm_col = []\n",
    "\n",
    "\n",
    "for i,r in summary_df.iterrows():\n",
    "    if len(summary_df.genomeID.loc[i]) < 3:\n",
    "        summary_df.genomeID.loc[i] = \"1%s\"%summary_df.genomeID.loc[i]\n",
    "    if summary_df.genomeID.loc[i] in merged_checkm.keys():\n",
    "        count += 1\n",
    "        checkm_col.append(merged_checkm[summary_df.genomeID.loc[i]])\n",
    "    else:\n",
    "        checkm_col.append(\"NA\")\n",
    "\n",
    "summary_df[\"completeness\"] = checkm_col\n",
    "\n",
    "summary_df = summary_df.replace(\"NA\",0) #zero completeness means the genome did not get scaffolded\n",
    "\n",
    "summary_df[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/tiago/anaconda2/lib/python2.7/site-packages/matplotlib/axes/_axes.py:6462: UserWarning: The 'normed' kwarg is deprecated, and has been replaced by the 'density' kwarg.\n",
      "  warnings.warn(\"The 'normed' kwarg is deprecated, and has been \"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7f9c5bb37d90>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAEKCAYAAAAcgp5RAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAEI5JREFUeJzt3X+MpVV9x/H3RxZUsLr8GAjuggvJ1l+0Ck4UtFoDphU0QhpUhJTVbrJNQxV/RMX+Mv2jidTfbQ3NFpCVIsWuRNAaKV1BYxpXZ4EK64pQtMvKyo4RUGxTQL7943m2nS4zzOz9wTBn36/k5j7Puee5z3n2zH7mzLn3npuqQpLUrqcsdgMkSeNl0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGGfSS1DiDXpIat2yxGwBw2GGH1apVqxa7GZK0pGzZsuUnVTUxX715gz7JpcDrgV1VdVxfdghwFbAK+CHwpqq6L0mATwKnAf8JvLWqbprvHKtWrWJqamq+apKkGZL8x0LqLWTq5jLgtXuUXQBsqqrVwKZ+H+BUYHV/WwdctJBGSJLGZ96gr6qvAz/do/h0YEO/vQE4Y0b5Z6rzTWB5kiNH1VhJ0t4b9MXYI6pqJ0B/f3hfvgK4e0a9HX3ZYyRZl2QqydT09PSAzZAkzWfU77rJLGWzroNcVeurarKqJicm5n0tQZI0oEGD/t7dUzL9/a6+fAdw1Ix6K4F7Bm+eJGlYgwb9tcCafnsNcM2M8nPTORF4YPcUjyRpcSzk7ZVXAq8GDkuyA/gg8CHgc0nWAtuBN/bVv0z31so76d5e+bYxtFmStBfmDfqqesscD50yS90Czhu2UZKk0XEJBElq3JNiCQRJWkyf3bx90c599suOHvs5HNFLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktS4oYI+ybuSbE1yW5IrkzwtyTFJNie5I8lVSQ4YVWMlSXtv4KBPsgJ4BzBZVccB+wFnARcCH6+q1cB9wNpRNFSSNJhhp26WAU9Psgw4ENgJnAxs7B/fAJwx5DkkSUMYOOir6kfAR4DtdAH/ALAFuL+qHumr7QBWDNtISdLghpm6ORg4HTgGeDZwEHDqLFVrjuPXJZlKMjU9PT1oMyRJ8xhm6uY1wA+qarqqHgauBl4OLO+ncgBWAvfMdnBVra+qyaqanJiYGKIZkqTHM0zQbwdOTHJgkgCnAN8FbgDO7OusAa4ZromSpGEMM0e/me5F15uAW/vnWg+8H3h3kjuBQ4FLRtBOSdKAls1fZW5V9UHgg3sU3wW8dJjnlSSNjp+MlaTGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktS4oYI+yfIkG5N8L8m2JCclOSTJ9Unu6O8PHlVjJUl7b9gR/SeBr1TV84AXAduAC4BNVbUa2NTvS5IWycBBn+SZwKuASwCq6qGquh84HdjQV9sAnDFsIyVJgxtmRH8sMA18OsnNSS5OchBwRFXtBOjvDx9BOyVJAxom6JcBJwAXVdXxwC/Yi2maJOuSTCWZmp6eHqIZkqTHM0zQ7wB2VNXmfn8jXfDfm+RIgP5+12wHV9X6qpqsqsmJiYkhmiFJejwDB31V/Ri4O8lz+6JTgO8C1wJr+rI1wDVDtVCSNJRlQx7/duCKJAcAdwFvo/vl8bkka4HtwBuHPIckaQhDBX1V3QJMzvLQKcM8ryRpdPxkrCQ1zqCXpMYZ9JLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGDR30SfZLcnOSL/X7xyTZnOSOJFclOWD4ZkqSBjWKEf35wLYZ+xcCH6+q1cB9wNoRnEOSNKChgj7JSuB1wMX9foCTgY19lQ3AGcOcQ5I0nGFH9J8A3gc82u8fCtxfVY/0+zuAFbMdmGRdkqkkU9PT00M2Q5I0l4GDPsnrgV1VtWVm8SxVa7bjq2p9VU1W1eTExMSgzZAkzWPZEMe+AnhDktOApwHPpBvhL0+yrB/VrwTuGb6ZkqRBDTyir6oPVNXKqloFnAV8tarOAW4AzuyrrQGuGbqVkqSBjeN99O8H3p3kTro5+0vGcA5J0gINM3Xzv6rqRuDGfvsu4KWjeF5J0vD8ZKwkNc6gl6TGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEj+eKRxfTZzdsX7dxnv+zoRTu3JC2UI3pJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGGfSS1DiDXpIaZ9BLUuMGDvokRyW5Icm2JFuTnN+XH5Lk+iR39PcHj665kqS9NcyI/hHgPVX1fOBE4LwkLwAuADZV1WpgU78vSVokAwd9Ve2sqpv67Z8D24AVwOnAhr7aBuCMYRspSRrcSObok6wCjgc2A0dU1U7ofhkAh4/iHJKkwQwd9EmeAXweeGdV/WwvjluXZCrJ1PT09LDNkCTNYaigT7I/XchfUVVX98X3Jjmyf/xIYNdsx1bV+qqarKrJiYmJYZohSXocw7zrJsAlwLaq+tiMh64F1vTba4BrBm+eJGlYy4Y49hXA7wK3JrmlL/sj4EPA55KsBbYDbxyuiZKkYQwc9FX1DSBzPHzKoM8rSRotPxkrSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekho3zHr0khr02c3bF7sJGjFH9JLUOINekhpn0EtS4wx6SWqcL8YOYbFetDr7ZUcvynklLU2O6CWpcY7otVf8K0ZaehzRS1LjHNEvQX6gZd9gP2tUHNFLUuMc0UuPw1G1WuCIXpIa54heS4Ija2lwjuglqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS48YS9Elem+T2JHcmuWAc55AkLczIgz7JfsCngFOBFwBvSfKCUZ9HkrQw4xjRvxS4s6ruqqqHgH8ATh/DeSRJCzCOoF8B3D1jf0dfJklaBONYAiGzlNVjKiXrgHX97oNJbh/wfIcBPxnw2KXKa943eM37gHOGu+bnLKTSOIJ+B3DUjP2VwD17Vqqq9cD6YU+WZKqqJod9nqXEa943eM37hifimscxdfNtYHWSY5IcAJwFXDuG80iSFmDkI/qqeiTJHwLXAfsBl1bV1lGfR5K0MGNZpriqvgx8eRzPPYuhp3+WIK953+A17xvGfs2peszrpJKkhrgEgiQ1bkkHfetLLSQ5KskNSbYl2Zrk/L78kCTXJ7mjvz94sds6akn2S3Jzki/1+8ck2dxf81X9C/3NSLI8ycYk3+v7+6TW+znJu/qf69uSXJnkaa31c5JLk+xKctuMsln7NZ2/6vPsO0lOGFU7lmzQ7yNLLTwCvKeqng+cCJzXX+MFwKaqWg1s6vdbcz6wbcb+hcDH+2u+D1i7KK0an08CX6mq5wEvorv2Zvs5yQrgHcBkVR1H98aNs2ivny8DXrtH2Vz9eiqwur+tAy4aVSOWbNCzDyy1UFU7q+qmfvvndP/5V9Bd54a+2gbgjMVp4XgkWQm8Dri43w9wMrCxr9LUNSd5JvAq4BKAqnqoqu6n8X6mezPI05MsAw4EdtJYP1fV14Gf7lE8V7+eDnymOt8Elic5chTtWMpBv08ttZBkFXA8sBk4oqp2QvfLADh88Vo2Fp8A3gc82u8fCtxfVY/0+6319bHANPDpfrrq4iQH0XA/V9WPgI8A2+kC/gFgC233825z9evYMm0pB/2CllpoQZJnAJ8H3llVP1vs9oxTktcDu6pqy8ziWaq21NfLgBOAi6rqeOAXNDRNM5t+Xvp04Bjg2cBBdFMXe2qpn+cztp/zpRz0C1pqYalLsj9dyF9RVVf3xffu/pOuv9+1WO0bg1cAb0jyQ7rpuJPpRvjL+z/xob2+3gHsqKrN/f5GuuBvuZ9fA/ygqqar6mHgauDltN3Pu83Vr2PLtKUc9M0vtdDPTV8CbKuqj8146FpgTb+9BrjmiW7buFTVB6pqZVWtouvTr1bVOcANwJl9tdau+cfA3Ume2xedAnyXhvuZbsrmxCQH9j/nu6+52X6eYa5+vRY4t3/3zYnAA7uneIZWVUv2BpwGfB/4d+CPF7s9Y7i+36D70+07wC397TS6OetNwB39/SGL3dYxXf+rgS/128cC3wLuBP4ReOpit2/E1/piYKrv6y8AB7fez8CfA98DbgMuB57aWj8DV9K9BvEw3Yh97Vz9Sjd186k+z26le0fSSNrhJ2MlqXFLeepGkrQABr0kNc6gl6TGGfSS1DiDXpIaZ9Brn5fksiRnzlPnrUme/US1SRolg15amLfSfVRfWnIMej1pJTm3X5f735JcnuQ5STb1ZZuSHN3XuyzJRf3a/Xcl+c1+HfBtSS6b8XwPJvlokpv64ydmOedLknwtyZYk1yU5sh/tTwJXJLklydNnq9cff2OSC5N8K8n3k7yyL98vyYeTfLtv/+/35Ucm+Xr/vLcleWVf97J+/9Yk73oC/rnVssX+5Jg3b7PdgBcCtwOH9fuHAF8E1vT7vwd8od++jG5dnNAtlPUz4NfoBjJbgBf39Qo4p9/+M+BvZhx/JrA/8K/ARF/+Zrovtwe4kf6Tiguo99F++zTgX/rtdcCf9NtPpfsU7DHAe+g/1U23JvuvAC8Brp/xb7F8sfvD29K+jeXLwaUROBnYWFU/AaiqnyY5Cfid/vHLgb+cUf+LVVVJbgXurapbAZJsBVbRLR/xKHBVX//v6RbSmum5wHHA9d3yK+xH9/H1Pc1Xb/fzbunPDfBbwK/PeC3gWXRfMPFt4NJ+8bovVNUtSe4Cjk3y18A/Af88x7+RtCAGvZ6swvxLtM58/L/7+0dnbO/en+vnfM/nD7C1qk5aQNser97u8/9yxrkDvL2qrnvMkyWvovuilcuTfLiqPpPkRcBvA+cBb6L7C0YaiHP0erLaBLwpyaHQfc8m3XTJWf3j5wDf2MvnfAr/tzLi2bMcfzsw0f/lQJL9k7ywf+zndNMq89Wby3XAH/Qjd5L8apKDkjyHbv39v6NbqfSEJIcBT6mqzwN/SrdksTQwR/R6UqqqrUn+Avhakl8CN9N9x+ilSd5L941Mb9vLp/0F8MIkW+i+0ejNe5zzoX5q5a+SPIvu/8cngK108/h/m+S/gJPofmHMVm8uF9NN49zUL8s7TfcVcq8G3pvkYeBB4Fy6bxX6dJLdA7EP7OV1Sv+Pq1dqn5Hkwap6xmK3Q3qiOXUjSY1zRC9JjXNEL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhr3P8JQqqozLdYVAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary_plot = summary_df[summary_df.completeness != 0]\n",
    "\n",
    "sns.distplot( summary_plot[\"completeness\"], bins=10,kde=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(80, 3)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df = summary_df[summary_df.completeness >= 90]\n",
    "\n",
    "master_df = master_df[~master_df['genomeID'].isin([\"4C7\",\"4B8\",\"4C8\",\"3F3\",\"4B7\",\"1E3\",\"2D8\",\"1B8\",\"1C1\",\"3F1\",\"1C9\",\"4B2\",\"4A8\",\"2I9\",\"2H1\",\"2I6\",\"1D4\"])]\n",
    "\n",
    "master_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(81, 3)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#adding ISBB and ASIH to master_df\n",
    "\n",
    "df_ISBB = pd.DataFrame(columns=['collectionID', 'genomeID', 'completeness'])\n",
    "\n",
    "df_ISBB.loc[0] = [\"ISB3NOV948BCUL\",\"ISBB\",100]\n",
    "\n",
    "df_ASIH = pd.DataFrame(columns=['collectionID', 'genomeID', 'completeness'])\n",
    "\n",
    "df_ASIH.loc[0] = [\"ASI16JUL142CUL\",\"ASIH\",0]\n",
    "\n",
    "master_df = master_df.append(df_ISBB,ignore_index=True)\n",
    "\n",
    "master_df = master_df.append(df_ASIH,ignore_index=True)\n",
    "\n",
    "master_df = master_df[~master_df['genomeID'].isin([\"4A6\"])]\n",
    "\n",
    "master_df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Confirming match between master_df and genomes in scaffold folder**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "81\n"
     ]
    }
   ],
   "source": [
    "ls /home/tiago/Desktop/cyanet/cyanobiome/scaffolds/ | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "for item in master_df[\"genomeID\"]:\n",
    "    glob_file = glob.glob(\"/home/tiago/Desktop/cyanet/cyanobiome/scaffolds/%s.fasta\"%item)\n",
    "    if not glob_file:\n",
    "        print item"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Expanding master_df with counts**\n",
    "\n",
    "```\n",
    "First, process the biosynthetic gene clusters using the big-scape notebook!\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_draft_counts(query_genome):\n",
    "    input_handle = open(query_genome, \"rU\")\n",
    "    node_count = 0\n",
    "    gc_count = []\n",
    "    for record in SeqIO.parse(input_handle, \"fasta\"):\n",
    "        node_count += 1\n",
    "        gc_count.append(GC(record.seq))\n",
    "    gc_average = np.average(gc_count)\n",
    "    return node_count,gc_average\n",
    "    input_handle.close()\n",
    "    \n",
    "def get_count_cols(summary_df,query_list):\n",
    "    count1_list = []\n",
    "    count2_list = []\n",
    "    for i,r in summary_df.iterrows():\n",
    "        temp_list = summary_df.genomeID.loc[i].split(\"-\")\n",
    "        for item in temp_list:\n",
    "            paths = [s for s in query_list if item in s]\n",
    "        if len(paths) > 0:\n",
    "            size_list = []\n",
    "            gc_list = []\n",
    "            for fasta in paths:\n",
    "                node_count,gc_count = get_draft_counts(fasta)\n",
    "                size_list.append(node_count)\n",
    "                gc_list.append(gc_count)\n",
    "            minpos = size_list.index(min(size_list))\n",
    "            count1_list.append(min(size_list))\n",
    "            count2_list.append(round(gc_list[minpos],2))\n",
    "        else:\n",
    "            count1_list.append(\"NA\")\n",
    "            count2_list.append(\"NA\")\n",
    "    return count1_list,count2_list\n",
    "\n",
    "def get_bgc_cols(summary_df,query_list):\n",
    "    count1_list = []\n",
    "    count2_list = []\n",
    "    for i,r in summary_df.iterrows():\n",
    "        temp_list = summary_df.genomeID.loc[i].split(\"-\")\n",
    "        for item in temp_list:\n",
    "            paths = [s for s in query_list if item in s]\n",
    "        if len(paths) > 0:\n",
    "            for filename in paths:\n",
    "                code = os.path.basename(filename).split(\".\")[0]\n",
    "                if len(code) < 3:\n",
    "                    code = \"1%s\"%code\n",
    "                all_bgcs = len(glob.glob(\"/home/tiago/Desktop/cyanet/bigscape/all_bgcs/%s_*gbk\"%code))\n",
    "                incomp_bgcs = len(glob.glob(\"/home/tiago/Desktop/cyanet/bigscape/incomplete_inputs/%s_*gbk\"%code))\n",
    "        count1_list.append(all_bgcs)\n",
    "        count2_list.append(incomp_bgcs)\n",
    "    return count1_list,count2_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_list = glob.glob(\"/home/tiago/Desktop/cyanet/cyanobiome/scaffolds/*\")\n",
    "count1_col,count2_col = get_count_cols(master_df,query_list)\n",
    "master_df[\"#scaffolds\"] = count1_col\n",
    "master_df[\"GC\"] = count2_col\n",
    "all_count_list, inc_count_list= get_bgc_cols(master_df,query_list)\n",
    "comp_count_list = []\n",
    "for i,item in enumerate(all_count_list):\n",
    "    new_item = item - inc_count_list[i]\n",
    "    comp_count_list.append(new_item)\n",
    "master_df[\"#fragBGCs\"] = inc_count_list\n",
    "master_df[\"#completeBGCs\"] = comp_count_list\n",
    "master_df[\"#allBGCs\"] = all_count_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "master_df = master_df.reset_index(inplace=False,drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASG15JUL146CUL</td>\n",
       "      <td>3F2</td>\n",
       "      <td>99.09</td>\n",
       "      <td>1606</td>\n",
       "      <td>58.04</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASX22JUL142CUL</td>\n",
       "      <td>1E4</td>\n",
       "      <td>99.73</td>\n",
       "      <td>2557</td>\n",
       "      <td>55.18</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CobbledownCUL</td>\n",
       "      <td>4C1</td>\n",
       "      <td>99.18</td>\n",
       "      <td>1299</td>\n",
       "      <td>56.61</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>numC11CUL</td>\n",
       "      <td>4C4</td>\n",
       "      <td>94.38</td>\n",
       "      <td>3601</td>\n",
       "      <td>54.79</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>numC15BCUL</td>\n",
       "      <td>4C5</td>\n",
       "      <td>98.64</td>\n",
       "      <td>1877</td>\n",
       "      <td>56.32</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PAB16MAY116CUL</td>\n",
       "      <td>3F4</td>\n",
       "      <td>98.91</td>\n",
       "      <td>2691</td>\n",
       "      <td>52.85</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PAB18MAY119CUL</td>\n",
       "      <td>1E1</td>\n",
       "      <td>99.18</td>\n",
       "      <td>2004</td>\n",
       "      <td>50.19</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PAL23MAY131CUL</td>\n",
       "      <td>1I2</td>\n",
       "      <td>98.51</td>\n",
       "      <td>1422</td>\n",
       "      <td>51.47</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PAL24MAY135-contACUL</td>\n",
       "      <td>1I4</td>\n",
       "      <td>99.46</td>\n",
       "      <td>2228</td>\n",
       "      <td>62.72</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PAP25JUN122CUL</td>\n",
       "      <td>1D8</td>\n",
       "      <td>97.09</td>\n",
       "      <td>3292</td>\n",
       "      <td>54.75</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PhormidiumspChinaCUL</td>\n",
       "      <td>1D9</td>\n",
       "      <td>99.05</td>\n",
       "      <td>806</td>\n",
       "      <td>54.44</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PNG22APR061CUL</td>\n",
       "      <td>3F7</td>\n",
       "      <td>98.91</td>\n",
       "      <td>645</td>\n",
       "      <td>49.67</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ASF14JUL142ENV</td>\n",
       "      <td>1B1</td>\n",
       "      <td>97.06</td>\n",
       "      <td>2301</td>\n",
       "      <td>43.12</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ASG15JUL146ENV</td>\n",
       "      <td>1B6</td>\n",
       "      <td>99.37</td>\n",
       "      <td>1063</td>\n",
       "      <td>50.13</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ASG15JUL148ENV</td>\n",
       "      <td>1C4</td>\n",
       "      <td>96.97</td>\n",
       "      <td>2840</td>\n",
       "      <td>45.65</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ASI16JUL142ENV</td>\n",
       "      <td>1A2</td>\n",
       "      <td>95.97</td>\n",
       "      <td>5391</td>\n",
       "      <td>44.71</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ASI16JUL144ENV</td>\n",
       "      <td>1A3</td>\n",
       "      <td>99.18</td>\n",
       "      <td>5306</td>\n",
       "      <td>43.37</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ASI16JUL149ENV</td>\n",
       "      <td>1A7</td>\n",
       "      <td>99.64</td>\n",
       "      <td>238</td>\n",
       "      <td>48.60</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AST13JUL143ENV</td>\n",
       "      <td>1F2</td>\n",
       "      <td>98.51</td>\n",
       "      <td>989</td>\n",
       "      <td>46.36</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ASY22JUL141ENV</td>\n",
       "      <td>1C2</td>\n",
       "      <td>99.09</td>\n",
       "      <td>498</td>\n",
       "      <td>46.29</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GBB21MAR161ENV</td>\n",
       "      <td>4G2</td>\n",
       "      <td>90.26</td>\n",
       "      <td>6730</td>\n",
       "      <td>43.96</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GGG21MAR161ENV</td>\n",
       "      <td>4G3</td>\n",
       "      <td>98.82</td>\n",
       "      <td>1456</td>\n",
       "      <td>44.95</td>\n",
       "      <td>48</td>\n",
       "      <td>2</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NAC09DEC082ENV</td>\n",
       "      <td>1I7</td>\n",
       "      <td>99.73</td>\n",
       "      <td>883</td>\n",
       "      <td>45.21</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NAC18DEC082ENV</td>\n",
       "      <td>3A5</td>\n",
       "      <td>98.91</td>\n",
       "      <td>201</td>\n",
       "      <td>43.38</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PAB03APR065ENV</td>\n",
       "      <td>1H2</td>\n",
       "      <td>99.73</td>\n",
       "      <td>1646</td>\n",
       "      <td>46.49</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PAB05APR064ENV</td>\n",
       "      <td>1H4</td>\n",
       "      <td>99.46</td>\n",
       "      <td>253</td>\n",
       "      <td>38.31</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PAB07APR054ENV</td>\n",
       "      <td>1H5</td>\n",
       "      <td>99.59</td>\n",
       "      <td>783</td>\n",
       "      <td>44.45</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PAB07APR055ENV</td>\n",
       "      <td>1H6</td>\n",
       "      <td>98.14</td>\n",
       "      <td>15041</td>\n",
       "      <td>38.33</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PAB16MAY112ENV</td>\n",
       "      <td>2A8</td>\n",
       "      <td>97.03</td>\n",
       "      <td>5113</td>\n",
       "      <td>45.51</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PAB16MAY113ENV</td>\n",
       "      <td>2B7</td>\n",
       "      <td>98.23</td>\n",
       "      <td>5085</td>\n",
       "      <td>42.26</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>PAL24MAY135ENV</td>\n",
       "      <td>2H7</td>\n",
       "      <td>91.55</td>\n",
       "      <td>7178</td>\n",
       "      <td>41.10</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>PAP14JUN083ENV</td>\n",
       "      <td>3C6</td>\n",
       "      <td>100</td>\n",
       "      <td>1290</td>\n",
       "      <td>47.03</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>PAP15JUN091ENV</td>\n",
       "      <td>4E2</td>\n",
       "      <td>97.83</td>\n",
       "      <td>1986</td>\n",
       "      <td>43.29</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>PAP15JUN097ENV</td>\n",
       "      <td>3B2</td>\n",
       "      <td>97.28</td>\n",
       "      <td>1473</td>\n",
       "      <td>44.00</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>PAP23MAR121ENV</td>\n",
       "      <td>3B3</td>\n",
       "      <td>95.88</td>\n",
       "      <td>3750</td>\n",
       "      <td>42.37</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>PAP23MAR125ENV</td>\n",
       "      <td>3I5</td>\n",
       "      <td>99.91</td>\n",
       "      <td>898</td>\n",
       "      <td>36.95</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>PAP25JUN121ENV</td>\n",
       "      <td>3E6</td>\n",
       "      <td>95.15</td>\n",
       "      <td>2743</td>\n",
       "      <td>43.47</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>PAP25JUN122ENV</td>\n",
       "      <td>3A1</td>\n",
       "      <td>97.28</td>\n",
       "      <td>4619</td>\n",
       "      <td>43.39</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>PAP28JUN071ENV</td>\n",
       "      <td>3E2</td>\n",
       "      <td>92.53</td>\n",
       "      <td>3382</td>\n",
       "      <td>43.78</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>PLP20MAR122ENV</td>\n",
       "      <td>3E8</td>\n",
       "      <td>98.91</td>\n",
       "      <td>744</td>\n",
       "      <td>51.38</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>PNG01MAY071ENV</td>\n",
       "      <td>2E6</td>\n",
       "      <td>98.91</td>\n",
       "      <td>2645</td>\n",
       "      <td>44.68</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>PNG11DEC03AENV</td>\n",
       "      <td>3H1</td>\n",
       "      <td>98.64</td>\n",
       "      <td>1667</td>\n",
       "      <td>40.60</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>PNG12JUL0731ENV</td>\n",
       "      <td>3I6</td>\n",
       "      <td>98.64</td>\n",
       "      <td>3176</td>\n",
       "      <td>43.47</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>PNG12JUL075ENV</td>\n",
       "      <td>3I7</td>\n",
       "      <td>94.93</td>\n",
       "      <td>5827</td>\n",
       "      <td>42.97</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>PNG12JUL076ENV</td>\n",
       "      <td>3I8</td>\n",
       "      <td>98.91</td>\n",
       "      <td>1581</td>\n",
       "      <td>43.45</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>PNG13DEC032ENV</td>\n",
       "      <td>3H5</td>\n",
       "      <td>98.14</td>\n",
       "      <td>2753</td>\n",
       "      <td>42.45</td>\n",
       "      <td>38</td>\n",
       "      <td>2</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>PNG13JUL072ENV</td>\n",
       "      <td>4A1</td>\n",
       "      <td>98.37</td>\n",
       "      <td>3031</td>\n",
       "      <td>43.44</td>\n",
       "      <td>13</td>\n",
       "      <td>4</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>PNG13JUL078ENV</td>\n",
       "      <td>3B5</td>\n",
       "      <td>99.09</td>\n",
       "      <td>1094</td>\n",
       "      <td>37.65</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>PNG15JUL074ENV</td>\n",
       "      <td>4A3</td>\n",
       "      <td>92.39</td>\n",
       "      <td>5965</td>\n",
       "      <td>44.09</td>\n",
       "      <td>40</td>\n",
       "      <td>0</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>PNG16DEC032ENV</td>\n",
       "      <td>3G5</td>\n",
       "      <td>95.65</td>\n",
       "      <td>2719</td>\n",
       "      <td>43.99</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>PNG16JUL074ENV</td>\n",
       "      <td>4A5</td>\n",
       "      <td>98.01</td>\n",
       "      <td>5470</td>\n",
       "      <td>43.46</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>PNG21MAY053ENV</td>\n",
       "      <td>2E9</td>\n",
       "      <td>99.37</td>\n",
       "      <td>814</td>\n",
       "      <td>44.73</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>PRL23MAR111ENV</td>\n",
       "      <td>1F9</td>\n",
       "      <td>99.73</td>\n",
       "      <td>556</td>\n",
       "      <td>45.12</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>PRL23MAR113ENV</td>\n",
       "      <td>1G1</td>\n",
       "      <td>100</td>\n",
       "      <td>570</td>\n",
       "      <td>45.28</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>PRM24MAR111ENV</td>\n",
       "      <td>1G2</td>\n",
       "      <td>98.37</td>\n",
       "      <td>1686</td>\n",
       "      <td>42.82</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>PRM25MAR112ENV</td>\n",
       "      <td>1G6</td>\n",
       "      <td>98.91</td>\n",
       "      <td>593</td>\n",
       "      <td>51.60</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>SPB31JAN131ENV</td>\n",
       "      <td>3A2</td>\n",
       "      <td>98.91</td>\n",
       "      <td>203</td>\n",
       "      <td>43.83</td>\n",
       "      <td>23</td>\n",
       "      <td>8</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>VQD27MAR117ENV</td>\n",
       "      <td>3C2</td>\n",
       "      <td>95.11</td>\n",
       "      <td>7532</td>\n",
       "      <td>46.87</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>ISB3NOV948BCUL</td>\n",
       "      <td>ISBB</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>52.10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>ASI16JUL142CUL</td>\n",
       "      <td>ASIH</td>\n",
       "      <td>0</td>\n",
       "      <td>674</td>\n",
       "      <td>49.60</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>81 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            collectionID genomeID completeness  #scaffolds     GC  #fragBGCs  \\\n",
       "0         ASG15JUL146CUL      3F2        99.09        1606  58.04         10   \n",
       "1         ASX22JUL142CUL      1E4        99.73        2557  55.18         12   \n",
       "2          CobbledownCUL      4C1        99.18        1299  56.61         21   \n",
       "3              numC11CUL      4C4        94.38        3601  54.79          4   \n",
       "4             numC15BCUL      4C5        98.64        1877  56.32         15   \n",
       "5         PAB16MAY116CUL      3F4        98.91        2691  52.85          9   \n",
       "6         PAB18MAY119CUL      1E1        99.18        2004  50.19         23   \n",
       "7         PAL23MAY131CUL      1I2        98.51        1422  51.47         10   \n",
       "8   PAL24MAY135-contACUL      1I4        99.46        2228  62.72         11   \n",
       "9         PAP25JUN122CUL      1D8        97.09        3292  54.75          9   \n",
       "10  PhormidiumspChinaCUL      1D9        99.05         806  54.44         16   \n",
       "11        PNG22APR061CUL      3F7        98.91         645  49.67         22   \n",
       "12        ASF14JUL142ENV      1B1        97.06        2301  43.12         28   \n",
       "13        ASG15JUL146ENV      1B6        99.37        1063  50.13          8   \n",
       "14        ASG15JUL148ENV      1C4        96.97        2840  45.65         14   \n",
       "15        ASI16JUL142ENV      1A2        95.97        5391  44.71         23   \n",
       "16        ASI16JUL144ENV      1A3        99.18        5306  43.37         22   \n",
       "17        ASI16JUL149ENV      1A7        99.64         238  48.60          6   \n",
       "18        AST13JUL143ENV      1F2        98.51         989  46.36         14   \n",
       "19        ASY22JUL141ENV      1C2        99.09         498  46.29         16   \n",
       "20        GBB21MAR161ENV      4G2        90.26        6730  43.96          4   \n",
       "21        GGG21MAR161ENV      4G3        98.82        1456  44.95         48   \n",
       "22        NAC09DEC082ENV      1I7        99.73         883  45.21          5   \n",
       "23        NAC18DEC082ENV      3A5        98.91         201  43.38         17   \n",
       "24        PAB03APR065ENV      1H2        99.73        1646  46.49         11   \n",
       "25        PAB05APR064ENV      1H4        99.46         253  38.31         11   \n",
       "26        PAB07APR054ENV      1H5        99.59         783  44.45         13   \n",
       "27        PAB07APR055ENV      1H6        98.14       15041  38.33         13   \n",
       "28        PAB16MAY112ENV      2A8        97.03        5113  45.51          2   \n",
       "29        PAB16MAY113ENV      2B7        98.23        5085  42.26         16   \n",
       "..                   ...      ...          ...         ...    ...        ...   \n",
       "51        PAL24MAY135ENV      2H7        91.55        7178  41.10          3   \n",
       "52        PAP14JUN083ENV      3C6          100        1290  47.03          6   \n",
       "53        PAP15JUN091ENV      4E2        97.83        1986  43.29         30   \n",
       "54        PAP15JUN097ENV      3B2        97.28        1473  44.00         23   \n",
       "55        PAP23MAR121ENV      3B3        95.88        3750  42.37         13   \n",
       "56        PAP23MAR125ENV      3I5        99.91         898  36.95          9   \n",
       "57        PAP25JUN121ENV      3E6        95.15        2743  43.47         16   \n",
       "58        PAP25JUN122ENV      3A1        97.28        4619  43.39         18   \n",
       "59        PAP28JUN071ENV      3E2        92.53        3382  43.78          3   \n",
       "60        PLP20MAR122ENV      3E8        98.91         744  51.38         23   \n",
       "61        PNG01MAY071ENV      2E6        98.91        2645  44.68         23   \n",
       "62        PNG11DEC03AENV      3H1        98.64        1667  40.60         11   \n",
       "63       PNG12JUL0731ENV      3I6        98.64        3176  43.47         13   \n",
       "64        PNG12JUL075ENV      3I7        94.93        5827  42.97         13   \n",
       "65        PNG12JUL076ENV      3I8        98.91        1581  43.45         15   \n",
       "66        PNG13DEC032ENV      3H5        98.14        2753  42.45         38   \n",
       "67        PNG13JUL072ENV      4A1        98.37        3031  43.44         13   \n",
       "68        PNG13JUL078ENV      3B5        99.09        1094  37.65         18   \n",
       "69        PNG15JUL074ENV      4A3        92.39        5965  44.09         40   \n",
       "70        PNG16DEC032ENV      3G5        95.65        2719  43.99         28   \n",
       "71        PNG16JUL074ENV      4A5        98.01        5470  43.46         15   \n",
       "72        PNG21MAY053ENV      2E9        99.37         814  44.73         16   \n",
       "73        PRL23MAR111ENV      1F9        99.73         556  45.12          5   \n",
       "74        PRL23MAR113ENV      1G1          100         570  45.28          4   \n",
       "75        PRM24MAR111ENV      1G2        98.37        1686  42.82          7   \n",
       "76        PRM25MAR112ENV      1G6        98.91         593  51.60         20   \n",
       "77        SPB31JAN131ENV      3A2        98.91         203  43.83         23   \n",
       "78        VQD27MAR117ENV      3C2        95.11        7532  46.87         23   \n",
       "79        ISB3NOV948BCUL     ISBB          100           2  52.10          0   \n",
       "80        ASI16JUL142CUL     ASIH            0         674  49.60          0   \n",
       "\n",
       "    #completeBGCs  #allBGCs  \n",
       "0               0        10  \n",
       "1              18        30  \n",
       "2               4        25  \n",
       "3               2         6  \n",
       "4               0        15  \n",
       "5               0         9  \n",
       "6               1        24  \n",
       "7               4        14  \n",
       "8               0        11  \n",
       "9               0         9  \n",
       "10              2        18  \n",
       "11              7        29  \n",
       "12              1        29  \n",
       "13              8        16  \n",
       "14              0        14  \n",
       "15              0        23  \n",
       "16              0        22  \n",
       "17              8        14  \n",
       "18              0        14  \n",
       "19             11        27  \n",
       "20              0         4  \n",
       "21              2        50  \n",
       "22             12        17  \n",
       "23             11        28  \n",
       "24              2        13  \n",
       "25              3        14  \n",
       "26              3        16  \n",
       "27              0        13  \n",
       "28              0         2  \n",
       "29              0        16  \n",
       "..            ...       ...  \n",
       "51              0         3  \n",
       "52             11        17  \n",
       "53              0        30  \n",
       "54              3        26  \n",
       "55              0        13  \n",
       "56              2        11  \n",
       "57              0        16  \n",
       "58              2        20  \n",
       "59              0         3  \n",
       "60              7        30  \n",
       "61              0        23  \n",
       "62              0        11  \n",
       "63              2        15  \n",
       "64              0        13  \n",
       "65              4        19  \n",
       "66              2        40  \n",
       "67              4        17  \n",
       "68              0        18  \n",
       "69              0        40  \n",
       "70              0        28  \n",
       "71              3        18  \n",
       "72              5        21  \n",
       "73              6        11  \n",
       "74              6        10  \n",
       "75              0         7  \n",
       "76             12        32  \n",
       "77              8        31  \n",
       "78              0        23  \n",
       "79              0         0  \n",
       "80              0         0  \n",
       "\n",
       "[81 rows x 8 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Correcting ISBB and ASIH number of BGCs**\n",
    "\n",
    "Since ISBB was in a separate folder, the total BGCs counted as zero when it is indeed 11 complete BGCs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>ISB3NOV948BCUL</td>\n",
       "      <td>ISBB</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>52.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      collectionID genomeID completeness  #scaffolds    GC  #fragBGCs  \\\n",
       "79  ISB3NOV948BCUL     ISBB          100           2  52.1          0   \n",
       "\n",
       "    #completeBGCs  #allBGCs  \n",
       "79              0         0  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df[master_df.genomeID == \"ISBB\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>ASI16JUL142CUL</td>\n",
       "      <td>ASIH</td>\n",
       "      <td>0</td>\n",
       "      <td>674</td>\n",
       "      <td>49.6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      collectionID genomeID completeness  #scaffolds    GC  #fragBGCs  \\\n",
       "80  ASI16JUL142CUL     ASIH            0         674  49.6          0   \n",
       "\n",
       "    #completeBGCs  #allBGCs  \n",
       "80              0         0  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df[master_df.genomeID == \"ASIH\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>ISB3NOV948BCUL</td>\n",
       "      <td>ISBB</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>52.1</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      collectionID genomeID completeness  #scaffolds    GC  #fragBGCs  \\\n",
       "79  ISB3NOV948BCUL     ISBB          100           2  52.1          0   \n",
       "\n",
       "    #completeBGCs  #allBGCs  \n",
       "79             11        11  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df.at[79,'#completeBGCs'] = 11\n",
    "master_df.at[79,'#allBGCs'] = 11\n",
    "\n",
    "master_df[master_df.genomeID == \"ISBB\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>ASI16JUL142CUL</td>\n",
       "      <td>ASIH</td>\n",
       "      <td>98.91</td>\n",
       "      <td>674</td>\n",
       "      <td>49.6</td>\n",
       "      <td>5</td>\n",
       "      <td>41</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      collectionID genomeID completeness  #scaffolds    GC  #fragBGCs  \\\n",
       "80  ASI16JUL142CUL     ASIH        98.91         674  49.6          5   \n",
       "\n",
       "    #completeBGCs  #allBGCs  \n",
       "80             41        46  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df.at[80,'#completeBGCs'] = 41\n",
    "master_df.at[80,'#allBGCs'] = 46\n",
    "master_df.at[80,'#fragBGCs'] = 5\n",
    "master_df.at[80,'completeness'] = 98.91\n",
    "\n",
    "master_df[master_df.genomeID == \"ASIH\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Finding GC cutoff using high GC genomes**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_gc = master_df[master_df[\"GC\"] >= 55]\n",
    "\n",
    "print len(df_gc)\n",
    "\n",
    "df_gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p /home/tiago/Desktop/cyanet/cyanobiome/quast/\n",
    "\n",
    "if not os.path.exists(\"/home/tiago/Desktop/cyanet/cyanobiome/quast/4B7.fasta\"):\n",
    "    for i,r in df_gc.iterrows():\n",
    "        subprocess.call(\"cp /home/tiago/Desktop/cyanet/cyanobiome/scaffolds/%s.fasta /home/tiago/Desktop/cyanet/cyanobiome/quast/\"%df_gc.genomeID.loc[i],shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls /home/tiago/Desktop/cyanet/cyanobiome/quast/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if not os.path.exists(\"/home/tiago/Desktop/cyanet/cyanobiome/quast/report.pdf\"):\n",
    "    !/home/tiago/quast-5.0.0/quast.py /home/tiago/Desktop/cyanet/cyanobiome/quast/*.fasta -o /home/tiago/Desktop/cyanet/cyanobiome/quast/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "According to the report above, the maximum GC should be set for 58%\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir /home/tiago/Desktop/cyanet/cyanobiome/final_scaffolds/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "First, filter all scaffolds with 1kb to obtain scaffolds size, then proceed below\n",
    "\n",
    "ls -lh /home/tiago/Desktop/cyanet/cyanobiome/final_scaffolds/ > /home/tiago/Desktop/cyanet/cyanobiome/1kb_scf_sizes.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1A2': 11.0,\n",
       " '1A3': 13.0,\n",
       " '1A7': 8.9,\n",
       " '1B1': 11.0,\n",
       " '1B6': 7.7,\n",
       " '1C2': 8.5,\n",
       " '1C4': 7.2,\n",
       " '1D4': 13.0,\n",
       " '1D8': 7.6,\n",
       " '1D9': 6.0,\n",
       " '1E1': 11.0,\n",
       " '1E4': 8.6,\n",
       " '1F2': 7.0,\n",
       " '1F9': 8.1,\n",
       " '1G1': 5.1,\n",
       " '1G2': 5.5,\n",
       " '1G6': 8.1,\n",
       " '1H2': 8.2,\n",
       " '1H4': 8.1,\n",
       " '1H5': 8.3,\n",
       " '1H6': 17.0,\n",
       " '1I2': 5.7,\n",
       " '1I4': 6.2,\n",
       " '1I7': 9.2,\n",
       " '2A8': 7.3,\n",
       " '2A9': 6.7,\n",
       " '2B3': 9.8,\n",
       " '2B6': 12.0,\n",
       " '2B7': 14.0,\n",
       " '2B9': 7.5,\n",
       " '2C1': 8.0,\n",
       " '2C2': 8.7,\n",
       " '2C4': 7.7,\n",
       " '2C9': 11.0,\n",
       " '2D1': 11.0,\n",
       " '2D2': 12.0,\n",
       " '2E6': 8.8,\n",
       " '2E9': 7.6,\n",
       " '2F4': 9.9,\n",
       " '2F5': 8.2,\n",
       " '2G3': 11.0,\n",
       " '2G4': 8.2,\n",
       " '2G5': 8.6,\n",
       " '2G6': 6.9,\n",
       " '2G7': 13.0,\n",
       " '2G8': 9.0,\n",
       " '2H1': 20.0,\n",
       " '2H7': 9.4,\n",
       " '2I5': 11.0,\n",
       " '2I6': 18.0,\n",
       " '2I9': 20.0,\n",
       " '3A1': 12.0,\n",
       " '3A2': 8.2,\n",
       " '3A5': 8.1,\n",
       " '3B2': 9.0,\n",
       " '3B3': 11.0,\n",
       " '3B5': 12.0,\n",
       " '3C2': 8.4,\n",
       " '3C4': 8.9,\n",
       " '3C6': 7.3,\n",
       " '3E2': 6.1,\n",
       " '3E6': 8.8,\n",
       " '3E8': 8.3,\n",
       " '3F2': 7.1,\n",
       " '3F4': 7.9,\n",
       " '3F7': 8.3,\n",
       " '3G5': 9.0,\n",
       " '3H1': 8.5,\n",
       " '3H5': 8.2,\n",
       " '3I5': 11.0,\n",
       " '3I6': 11.0,\n",
       " '3I7': 9.8,\n",
       " '3I8': 7.8,\n",
       " '4A1': 8.9,\n",
       " '4A3': 14.0,\n",
       " '4A5': 12.0,\n",
       " '4C1': 5.2,\n",
       " '4C4': 3.3,\n",
       " '4C5': 5.5,\n",
       " '4D6': 6.0,\n",
       " '4E2': 9.0,\n",
       " '4G2': 8.2,\n",
       " '4G3': 9.7,\n",
       " 'ASIH': 9.7,\n",
       " 'ISBB': 8.1}"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "size_df = pd.read_csv('/home/tiago/Desktop/cyanet/cyanobiome/1kb_scf_sizes.txt',sep=',',header=None)\n",
    "size_dict = pd.Series(size_df[1].values,index=size_df[0]).to_dict()\n",
    "\n",
    "size_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1A3',\n",
       " '1H6',\n",
       " '2H1',\n",
       " '1D4',\n",
       " '4A3',\n",
       " '2G7',\n",
       " '2D2',\n",
       " '2B7',\n",
       " '2B6',\n",
       " '3B5',\n",
       " '4A5',\n",
       " '3A1',\n",
       " '2I9',\n",
       " '2I6']"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "threekb_list = []\n",
    "\n",
    "for key in size_dict:\n",
    "    if size_dict[key] > 11:\n",
    "        threekb_list.append(key)\n",
    "        \n",
    "threekb_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "454 contigs renamed\n",
      "659 contigs renamed\n",
      "849 contigs renamed\n",
      "1496 contigs renamed\n",
      "1204 contigs renamed\n",
      "1509 contigs renamed\n",
      "1398 contigs renamed\n",
      "1046 contigs renamed\n",
      "245 contigs renamed\n",
      "1787 contigs renamed\n",
      "332 contigs renamed\n",
      "399 contigs renamed\n",
      "1462 contigs renamed\n",
      "265 contigs renamed\n",
      "1267 contigs renamed\n",
      "2584 contigs renamed\n",
      "1374 contigs renamed\n",
      "183 contigs renamed\n",
      "688 contigs renamed\n",
      "246 contigs renamed\n",
      "3478 contigs renamed\n",
      "777 contigs renamed\n",
      "110 contigs renamed\n",
      "164 contigs renamed\n",
      "279 contigs renamed\n",
      "192 contigs renamed\n",
      "263 contigs renamed\n",
      "1436 contigs renamed\n",
      "3058 contigs renamed\n",
      "1655 contigs renamed\n",
      "2028 contigs renamed\n",
      "1833 contigs renamed\n",
      "471 contigs renamed\n",
      "250 contigs renamed\n",
      "687 contigs renamed\n",
      "3347 contigs renamed\n",
      "184 contigs renamed\n",
      "1595 contigs renamed\n",
      "1186 contigs renamed\n",
      "1513 contigs renamed\n",
      "318 contigs renamed\n",
      "5156 contigs renamed\n",
      "339 contigs renamed\n",
      "3628 contigs renamed\n",
      "334 contigs renamed\n",
      "3842 contigs renamed\n",
      "590 contigs renamed\n",
      "1149 contigs renamed\n",
      "1348 contigs renamed\n",
      "191 contigs renamed\n",
      "2451 contigs renamed\n",
      "4053 contigs renamed\n",
      "290 contigs renamed\n",
      "1452 contigs renamed\n",
      "1018 contigs renamed\n",
      "2567 contigs renamed\n",
      "563 contigs renamed\n",
      "1884 contigs renamed\n",
      "772 contigs renamed\n",
      "2452 contigs renamed\n",
      "387 contigs renamed\n",
      "1438 contigs renamed\n",
      "1134 contigs renamed\n",
      "1790 contigs renamed\n",
      "3188 contigs renamed\n",
      "511 contigs renamed\n",
      "835 contigs renamed\n",
      "1106 contigs renamed\n",
      "790 contigs renamed\n",
      "1522 contigs renamed\n",
      "2144 contigs renamed\n",
      "605 contigs renamed\n",
      "384 contigs renamed\n",
      "198 contigs renamed\n",
      "39 contigs renamed\n",
      "1028 contigs renamed\n",
      "205 contigs renamed\n",
      "132 contigs renamed\n",
      "2364 contigs renamed\n",
      "3 contigs renamed\n",
      "25 contigs renamed\n"
     ]
    }
   ],
   "source": [
    "def select_scaffolds(fasta_file,strain,outfile,cutoff,threekb_list):\n",
    "    strain = os.path.basename(fasta_file).split(\".\")[0]\n",
    "    if strain in threekb_list:\n",
    "        size_cutoff = 3000\n",
    "    else:\n",
    "        size_cutoff = 1000\n",
    "    new_file = []\n",
    "    gc_count = []\n",
    "    input_handle = open(fasta_file, \"rU\")\n",
    "    node_count = 1\n",
    "    for record in SeqIO.parse(input_handle, \"fasta\"):\n",
    "        if len(record) >= size_cutoff:\n",
    "            if GC(record.seq) < cutoff:\n",
    "                new_id = \"%s_NODE_%s\"%(strain,node_count)\n",
    "                record.id = str(new_id)\n",
    "                record.description = \"\"\n",
    "                new_file.append(record)\n",
    "                node_count += 1\n",
    "                gc_count.append(GC(record.seq))\n",
    "    output_handle = open(outfile, \"w\")\n",
    "    SeqIO.write(new_file, output_handle, \"fasta\")\n",
    "    output_handle.close()\n",
    "    input_handle.close()\n",
    "    print \"%s contigs renamed\"%(node_count)\n",
    "    gc_average = np.average(gc_count)\n",
    "    return node_count,round(gc_average,2)\n",
    "\n",
    "new_size_col = []\n",
    "new_gc_col = []\n",
    "\n",
    "for strain in list(master_df.genomeID):\n",
    "    fasta_file = glob.glob(\"/home/tiago/Desktop/cyanet/cyanobiome/scaffolds/%s.fasta\"%strain)[0]\n",
    "    strain = \"SIO\"+strain\n",
    "    outfile = \"/home/tiago/Desktop/cyanet/cyanobiome/final_scaffolds/%s.fasta\"%(strain)\n",
    "    size,gc_perc = select_scaffolds(fasta_file,strain,outfile,58,threekb_list)\n",
    "    new_size_col.append(size)\n",
    "    new_gc_col.append(gc_perc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Updating node count in master_df**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASG15JUL146CUL</td>\n",
       "      <td>3F2</td>\n",
       "      <td>99.09</td>\n",
       "      <td>454</td>\n",
       "      <td>49.40</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASX22JUL142CUL</td>\n",
       "      <td>1E4</td>\n",
       "      <td>99.73</td>\n",
       "      <td>659</td>\n",
       "      <td>52.29</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CobbledownCUL</td>\n",
       "      <td>4C1</td>\n",
       "      <td>99.18</td>\n",
       "      <td>849</td>\n",
       "      <td>53.63</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>numC11CUL</td>\n",
       "      <td>4C4</td>\n",
       "      <td>94.38</td>\n",
       "      <td>1496</td>\n",
       "      <td>46.66</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>numC15BCUL</td>\n",
       "      <td>4C5</td>\n",
       "      <td>98.64</td>\n",
       "      <td>1204</td>\n",
       "      <td>52.29</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     collectionID genomeID completeness  #scaffolds     GC  #fragBGCs  \\\n",
       "0  ASG15JUL146CUL      3F2        99.09         454  49.40         10   \n",
       "1  ASX22JUL142CUL      1E4        99.73         659  52.29         12   \n",
       "2   CobbledownCUL      4C1        99.18         849  53.63         21   \n",
       "3       numC11CUL      4C4        94.38        1496  46.66          4   \n",
       "4      numC15BCUL      4C5        98.64        1204  52.29         15   \n",
       "\n",
       "   #completeBGCs  #allBGCs  \n",
       "0              0        10  \n",
       "1             18        30  \n",
       "2              4        25  \n",
       "3              2         6  \n",
       "4              0        15  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df[\"#scaffolds\"] = new_size_col\n",
    "master_df[\"GC\"] = new_gc_col\n",
    "\n",
    "master_df[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collectionID</th>\n",
       "      <th>genomeID</th>\n",
       "      <th>completeness</th>\n",
       "      <th>#scaffolds</th>\n",
       "      <th>GC</th>\n",
       "      <th>#fragBGCs</th>\n",
       "      <th>#completeBGCs</th>\n",
       "      <th>#allBGCs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASG15JUL146CUL</td>\n",
       "      <td>SIO3F2</td>\n",
       "      <td>99.09</td>\n",
       "      <td>454</td>\n",
       "      <td>49.40</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASX22JUL142CUL</td>\n",
       "      <td>SIO1E4</td>\n",
       "      <td>99.73</td>\n",
       "      <td>659</td>\n",
       "      <td>52.29</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CobbledownCUL</td>\n",
       "      <td>SIO4C1</td>\n",
       "      <td>99.18</td>\n",
       "      <td>849</td>\n",
       "      <td>53.63</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>numC11CUL</td>\n",
       "      <td>SIO4C4</td>\n",
       "      <td>94.38</td>\n",
       "      <td>1496</td>\n",
       "      <td>46.66</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>numC15BCUL</td>\n",
       "      <td>SIO4C5</td>\n",
       "      <td>98.64</td>\n",
       "      <td>1204</td>\n",
       "      <td>52.29</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     collectionID genomeID completeness  #scaffolds     GC  #fragBGCs  \\\n",
       "0  ASG15JUL146CUL   SIO3F2        99.09         454  49.40         10   \n",
       "1  ASX22JUL142CUL   SIO1E4        99.73         659  52.29         12   \n",
       "2   CobbledownCUL   SIO4C1        99.18         849  53.63         21   \n",
       "3       numC11CUL   SIO4C4        94.38        1496  46.66          4   \n",
       "4      numC15BCUL   SIO4C5        98.64        1204  52.29         15   \n",
       "\n",
       "   #completeBGCs  #allBGCs  \n",
       "0              0        10  \n",
       "1             18        30  \n",
       "2              4        25  \n",
       "3              2         6  \n",
       "4              0        15  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_names = []\n",
    "\n",
    "for i,r in master_df.iterrows():\n",
    "    strain = \"SIO\" + str(master_df.genomeID.loc[i])\n",
    "    new_names.append(strain)\n",
    "    \n",
    "master_df[\"genomeID\"] = new_names\n",
    "\n",
    "master_df[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "master_df.to_csv(\"/home/tiago/Desktop/cyanet/cyanobiome/tables/master_df-TFL190425.txt\",sep=\"\\t\",index_label=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp /home/tiago/Desktop/cyanet/cyanobiome/tables/master_df-TFL190425.txt /home/tiago/Desktop/cyanet/genomics_pnas/outputs/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp -r /home/tiago/Desktop/cyanet/cyanobiome/final_scaffolds/ /home/tiago/Desktop/cyanet/genomics_pnas/outputs/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "Outputs: master_df-TFL190425.txt\n",
    "         final scaffolds\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
